---
title: "iNext EcoZUR alpha diversity"
author: "Kelsey Jesser"
date: "7/19/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Set up
```{r}
#load R libraries
library(tidyverse)
library(qiime2R)
library(phyloseq)
library(ggplot2)
library(lefse)
library(plyr)
library(dplyr)
library(tibble)
library(data.table)
library(ggsci)
library(vegan)
library(lefse)
library(iNEXT)

#setwd
setwd("~/16S_EcoZUR_qiime2/symp_asymp/phyloseq_lefse")

#set theme
theme_set(theme_bw())
```

Import phyloseq object
```{r}
#import qiime files and convert to phyloseq object
phy<-qza_to_phyloseq("table_EcoZUR.qza", "rooted_tree_masked_aligned_rep_set_EcoZUR.qza", "taxonomy_EcoZUR.qza", "EcoZUR_meta_updated_pathotypes.tsv")

#filter taxa where phylum is ambiguous since these are likely artifacts
phy <- subset_taxa(phy, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))
phy

#filter chloroplast and mitochondrial sequences
phy<-subset_taxa(phy, (Class!="Cloroplast") | is.na(Class))
phy<-subset_taxa(phy, (Family!="mitochondria") | is.na(Class))

#filter samples with no metadata and samples
phy<-subset_samples(phy, sample_names(phy)!="B028")
phy<-subset_samples(phy, sample_names(phy)!="C001")
phy<-subset_samples(phy, sample_names(phy)!="C072")
phy<-subset_samples(phy, sample_names(phy)!="B071")
phy<-subset_samples(phy, sample_names(phy)!="R007")
```
Subset data
```{r}
#subset samples based on pathotype status
phy_pathpos<-subset_samples(phy, (pathotype_sum_genome=="1" | pathotype_sum_genome=="2"))
phy_pathneg<-subset_samples(phy, (pathotype_sum_genome=="0"))
                            
#separate based on pathotype and cc status
phy_pathpos_case<-subset_samples(phy_pathpos, cc=="case")
phy_pathpos_control<-subset_samples(phy_pathpos, cc=="control")
phy_pathneg_case<-subset_samples(phy_pathneg, cc=="case")
phy_pathneg_control<-subset_samples(phy_pathneg, cc=="control")
```

Convert to iNEXT-compatible formats
```{r}
#all 16S samples
phy_inext<-prepare_inext(phy)

#by pathotype status
phy_pathpos_inext<-prepare_inext(phy_pathpos)
phy_pathneg_inext<-prepare_inext(phy_pathneg)

#by pathotype and cc status
phy_pathpos_case_inext<-prepare_inext(phy_pathpos_case)
phy_pathpos_control_inext<-prepare_inext(phy_pathpos_control)
phy_pathneg_case_inext<-prepare_inext(phy_pathneg_case)
phy_pathneg_control_inext<-prepare_inext(phy_pathneg_control)

```

Run iNEXT
```{r}
#all 16S samples
phy_inext_out<-iNEXT(phy_inext, q=0, datatype="abundance")

#by pathotype status
phy_pathpos_inext_out<-iNEXT(phy_pathpos_inext, q=0, datatype="abundance")
phy_pathneg_inext_out<-iNEXT(phy_pathneg_inext, q=0, datatype="abundance")

#by pathotype and cc status
phy_pathpos_case_inext_out<-iNEXT(phy_pathpos_case_inext, q=0, datatype="abundance")
phy_pathpos_control_inext_out<-iNEXT(phy_pathpos_control_inext, q=0, datatype="abundance")
phy_pathneg_case_inext_out<-iNEXT(phy_pathneg_case_inext, q=0, datatype="abundance")
phy_pathneg_control_inext_out<-iNEXT(phy_pathneg_control_inext, q=0, datatype="abundance")

```
Plot rarefaction curves
```{r}
#all 16S samples
ggiNEXT(phy_inext_out, type=1, se=TRUE, facet.var="none", color.var="none", grey=FALSE)+theme(legend.position="none")+ggtitle("all samples")+labs(y="Diversity", x="ASVs")

#can also do others; not very useful without color-coded plots
```
Add cc and pathotype status labels to AsyEst outputs
```{r}
#extract metadata from phylose output and reformat
metadata<-sample_data(phy)
metadata<-tibble::rownames_to_column(metadata, "sample")
metadata<-data.frame(metadata)

#write AsyEst output to dataframe
AsyEst_phy<-phy_inext_out$AsyEst

#change AsyEst ouput column for merge
colnames(AsyEst_phy)[1]<-c("sample")

#merge AsyEst and metadata
AsyEst_phy_merged<-merge(AsyEst_phy, metadata, by = "sample")

#filter NAs
AsyEst_phy_merged<-filter(AsyEst_phy_merged, path_cc=="pathpos_case" | path_cc=="pathpos_control" | path_cc=="pathneg_case" | path_cc == "pathneg_control")

#make pathpos version for pathpos cc comparisons
AsyEst_phy_merged_pathpos<-filter(AsyEst_phy_merged, pathotype_status=="pathpos")
```


Plot estimated diversity (must add labels to AsyEst ouptuts)
```{r}
#by cc status
a<-ggplot(AsyEst_phy_merged, aes(y=Estimator, x=Diversity, color = cc))+geom_boxplot()
a

#by pathotype status
b<-ggplot(AsyEst_phy_merged, aes(y=Estimator, x=Diversity, color = pathotype_status))+geom_boxplot()
b

#by pathotype and cc status
c<-ggplot(AsyEst_phy_merged, aes(y=Estimator, x=Diversity, color = path_cc))+geom_boxplot()
c

d<-ggplot(AsyEst_phy_merged_pathpos, aes(y=Estimator, x=Diversity, color = cc))+geom_boxplot()
d
```

Mann Whitney U tests
```{r}
#subset data by diversity metric
asyest_rich<-filter(AsyEst_phy_merged, Diversity=="Species richness")
asyest_simp<-filter(AsyEst_phy_merged, Diversity=="Simpson diversity")
asyest_shan<-filter(AsyEst_phy_merged, Diversity=="Shannon diversity")

#subset shanon metric by case/control and pathotype status
asyest_shan_case<-filter(asyest_shan, cc=="case")
asyest_shan_control<-filter(asyest_shan, cc=="control")
asyest_shan_pathpos<-filter(asyest_shan, pathotype_status=="pathpos")
asyest_shan_pathneg<-filter(asyest_shan, pathotype_status=="pathneg")

asyest_shan_pathpos_case<-filter(asyest_shan, path_cc=="pathpos_case")
asyest_shan_pathpos_control<-filter(asyest_shan, path_cc=="pathpos_control")
asyest_shan_pathneg_case<-filter(asyest_shan, path_cc=="pathneg_case")
asyest_shan_pathneg_control<-filter(asyest_shan, path_cc=="pathneg_control")

#subset simpson metric by case/control and pathotype status
asyest_simp_case<-filter(asyest_simp, cc=="case")
asyest_simp_control<-filter(asyest_simp, cc=="control")
asyest_simp_pathpos<-filter(asyest_simp, pathotype_status=="pathpos")
asyest_simp_pathneg<-filter(asyest_simp, pathotype_status=="pathneg")

asyest_simp_pathpos_case<-filter(asyest_simp, path_cc=="pathpos_case")
asyest_simp_pathpos_control<-filter(asyest_simp, path_cc=="pathpos_control")
asyest_simp_pathneg_case<-filter(asyest_simp, path_cc=="pathneg_case")
asyest_simp_pathneg_control<-filter(asyest_simp, path_cc=="pathneg_control")

#subset observed/richness metric by case/control and pathotype status
asyest_rich_case<-filter(asyest_rich, cc=="case")
asyest_rich_control<-filter(asyest_rich, cc=="control")
asyest_rich_pathpos<-filter(asyest_rich, pathotype_status=="pathpos")
asyest_rich_pathneg<-filter(asyest_rich, pathotype_status=="pathneg")

asyest_rich_pathpos_case<-filter(asyest_rich, path_cc=="pathpos_case")
asyest_rich_pathpos_control<-filter(asyest_rich, path_cc=="pathpos_control")
asyest_rich_pathneg_case<-filter(asyest_rich, path_cc=="pathneg_case")
asyest_rich_pathneg_control<-filter(asyest_rich, path_cc=="pathneg_control")

#Mann Whitney U comparisons
#cc and path
wilcox.test(asyest_shan_case$Estimator, asyest_shan_control$Estimator)
wilcox.test(asyest_shan_pathpos$Estimator, asyest_shan_pathneg$Estimator)
wilcox.test(asyest_shan_pathpos_case$Estimator, asyest_shan_pathpos_control$Estimator)

wilcox.test(asyest_simp_case$Estimator, asyest_simp_control$Estimator)
wilcox.test(asyest_simp_pathpos$Estimator, asyest_simp_pathneg$Estimator)
wilcox.test(asyest_simp_pathpos_case$Estimator, asyest_simp_pathpos_control$Estimator)

wilcox.test(asyest_rich_case$Estimator, asyest_rich_control$Estimator)
wilcox.test(asyest_rich_pathpos$Estimator, asyest_rich_pathneg$Estimator)
wilcox.test(asyest_rich_pathpos_case$Estimator, asyest_rich_pathpos_control$Estimator)
```

