---
title: "Alpha diversity"
author: "Kelsey Jesser"
date: "2/24/2023"
output: pdf_document
---
#clear R env
```{r message=FALSE, warning=FALSE}
rm(list = ls())
```

#load libraries
```{r message=FALSE, warning=FALSE}
library(Nonpareil)
library(ggplot2)
library(ggpubr)
library(cowplot)
library(ggthemr)
library(dplyr)
library(tidyverse)
library(reshape2)
library(qiime2R)
library(tibble)
library(data.table)
library(ggsci)
library(iNEXT)
library(phyloseq)
library(ggpubr)
library(metagMisc)
library(ggplot2)
library(microViz)
```

#set theme
```{r message=FALSE, warning=FALSE}
ggthemr('fresh')
```

#iNEXT (16S data)
#Import qiime2 files as 16S phyloseq object
```{r message=FALSE, warning=FALSE}
phy<-qza_to_phyloseq("table_EcoZUR.qza", 
                     "rooted_tree_masked_aligned_rep_set_EcoZUR.qza", 
                     "taxonomy_EcoZUR.qza", 
                     "EcoZUR_meta_updated_pathotypes.tsv")
```

#filter phyloseq object
```{r include=FALSE}
#remove taxa where phylum is ambiguous (these are likely artifacts)
phy <- subset_taxa(phy, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))

#remove chloroplast and mitochondrial sequences
phy<-subset_taxa(phy, (Class!="Cloroplast") | is.na(Class))
phy<-subset_taxa(phy, (Family!="mitochondria") | is.na(Class))

#remove rotavirus-positive samples
phy<- subset_samples(phy, rota!="1")

#remove samples where isolate genome was contaminated
phy<-subset_samples(phy, contaminated!="1")

#remove E. coli
phy_Ec_removed<-subset_taxa(phy, (Species!="coli") | is.na(Class))

#add diarrhea and infection status metadata to phyloseq object
meta<-read.csv(file = "16S_metadata_DEC_diarrhea.csv")
phy<-ps_join(x = phy, y = meta, type = "inner", match_sample_names="SampleID")
phy_Ec_removed<-ps_join(x = phy_Ec_removed, y = meta, type = "inner", match_sample_names="SampleID")
```

#subset data by DEC infection and diarrhea status
```{r message=FALSE, warning=FALSE}
#with E. coli
#subset samples based on DEC infection status 
phy_Inf<-subset_samples(phy, (DEC_infection=="Infected"))
phy_Uninf<-subset_samples(phy, (DEC_infection=="Uninfected"))
                            
#subset samples based on diarrhea (case/control) and DEC infection status 
phy_Inf_case<-subset_samples(phy_Inf, cc=="Case")
phy_Inf_control<-subset_samples(phy_Inf, cc=="Control")
phy_Uninf_case<-subset_samples(phy_Uninf, cc=="Case")
phy_Uninf_control<-subset_samples(phy_Uninf, cc=="Control")

#without E. coli
phy_Ec_removed_Inf<-subset_samples(phy_Ec_removed, (DEC_infection=="Infected"))
phy_Ec_removed_Uninf<-subset_samples(phy_Ec_removed, (DEC_infection=="Uninfected"))
                            
#subset samples based on diarrhea (case/control) and DEC infection status 
phy_Ec_removed_Inf_case<-subset_samples(phy_Ec_removed_Inf, cc=="Case")
phy_Ec_removed_Inf_control<-subset_samples(phy_Ec_removed_Inf, cc=="Control")
phy_Ec_removed_Uninf_case<-subset_samples(phy_Ec_removed_Uninf, cc=="Case")
phy_Ec_removed_Uninf_control<-subset_samples(phy_Ec_removed_Uninf, cc=="Control")
```

#convert to iNEXT compatible formats
```{r message=FALSE, warning=FALSE}
#with  E. coli
#all samples
phy_inext<-prepare_inext(phy)

#by DEC infection status
phy_Inf_inext<-prepare_inext(phy_Inf)
phy_Uninf_inext<-prepare_inext(phy_Uninf)

#by diarrhea and DEC infection status
phy_Inf_case_inext<-prepare_inext(phy_Inf_case)
phy_Inf_control_inext<-prepare_inext(phy_Inf_control)
phy_Uninf_case_inext<-prepare_inext(phy_Uninf_case)
phy_Uninf_control_inext<-prepare_inext(phy_Uninf_control)

#without E. coli
phy_Ec_removed_inext<-prepare_inext(phy_Ec_removed)

#by DEC infection status
phy_Ec_removed_Inf_inext<-prepare_inext(phy_Ec_removed_Inf)
phy_Ec_removed_Uninf_inext<-prepare_inext(phy_Ec_removed_Uninf)

#by diarrhea and DEC infection status
phy_Ec_removed_Inf_case_inext<-prepare_inext(phy_Ec_removed_Inf_case)
phy_Ec_removed_Inf_control_inext<-prepare_inext(phy_Ec_removed_Inf_control)
phy_Ec_removed_Uninf_case_inext<-prepare_inext(phy_Ec_removed_Uninf_case)
phy_Ec_removed_Uninf_control_inext<-prepare_inext(phy_Ec_removed_Uninf_control)
```

#Run iNEXT
```{r message=FALSE, warning=FALSE}
#with E. coli
#by diarrhea and DEC infection status
phy_Inf_case_inext_out<-iNEXT(phy_Inf_case_inext, q=0, datatype="abundance")
phy_Inf_control_inext_out<-iNEXT(phy_Inf_control_inext, q=0, datatype="abundance")
phy_Uninf_case_inext_out<-iNEXT(phy_Uninf_case_inext, q=0, datatype="abundance")
phy_Uninf_control_inext_out<-iNEXT(phy_Uninf_control_inext, q=0, datatype="abundance")

#without E. coli
#by diarrhea and DEC infection status
phy_Ec_removed_Inf_case_inext_out<-iNEXT(phy_Ec_removed_Inf_case_inext, q=0, datatype="abundance")
phy_Ec_removed_Inf_control_inext_out<-iNEXT(phy_Ec_removed_Inf_control_inext, q=0, datatype="abundance")
phy_Ec_removed_Uninf_case_inext_out<-iNEXT(phy_Ec_removed_Uninf_case_inext, q=0, datatype="abundance")
phy_Ec_removed_Uninf_control_inext_out<-iNEXT(phy_Ec_removed_Uninf_control_inext, q=0, datatype="abundance")

```

Plot rarefaction curves
```{r message=FALSE, warning=FALSE}

rare<- ggiNEXT(phy_inext_out, type=1, se=TRUE, facet.var="none", color.var="none", grey=TRUE)+
  theme(legend.position="none")+
  ggtitle("All samples")+
  labs(y="Diversity", x="ASVs")
rare
```

#Format data to plot by diarrhea and DEC infection status
```{r message=FALSE, warning=FALSE}
#with E. coli
#extract metadata from phyloseq and reformat
metadata<-sample_data(phy)
metadata<-tibble::rownames_to_column(metadata, "sample")
metadata<-data.frame(metadata)

#write AsyEst output to dataframe
AsyEst_phy<-phy_inext_out$AsyEst

#change AsyEst ouput column for merge
colnames(AsyEst_phy)[1]<-c("sample")

#merge AsyEst and metadata
AsyEst_phy_merged<-AsyEst_phy %>% 
  merge(metadata, by = "sample" )%>%
  filter(Infection_diarrhea=="Infected_Case" | Infection_diarrhea=="Infected_Control" | Infection_diarrhea=="Uninfected_Case" | Infection_diarrhea == "Uninfected_Control")%>%
  select(c(sample, Diversity, Estimator, Diarrhea, DEC_infection, Infection_diarrhea))

#without E. coli
#extract metadata from phyloseq and reformat
metadata<-sample_data(phy_Ec_removed)
metadata<-tibble::rownames_to_column(metadata, "sample")
metadata<-data.frame(metadata)

#write AsyEst output to dataframe
AsyEst_phy_Ec_removed<-phy_Ec_removed_inext_out$AsyEst

#change AsyEst ouput column for merge
colnames(AsyEst_phy_Ec_removed)[1]<-c("sample")

#merge AsyEst and metadata
AsyEst_phy_Ec_removed_merged<-AsyEst_phy_Ec_removed %>% 
  merge(metadata, by = "sample" )%>%
  filter(Infection_diarrhea=="Infected_Case" | Infection_diarrhea=="Infected_Control" | Infection_diarrhea=="Uninfected_Case" | Infection_diarrhea == "Uninfected_Control")%>%
  select(c(sample, Diversity, Estimator, Diarrhea, DEC_infection, Infection_diarrhea))
```

#Boxplots with E. coli
```{r message=FALSE, warning=FALSE}
#shannon diversity
AsyEst_phy_merged_Shanon<- subset(AsyEst_phy_merged, Diversity=="Shannon diversity")

Shan<-ggplot(AsyEst_phy_merged_Shanon, aes(y=Estimator, x=DEC_infection, fill=Infection_diarrhea))+
  geom_boxplot()+
  ylab("Shannon index")+
  stat_compare_means(method="wilcox.test", label.y=62.5, size=3.5)+
  scale_x_discrete(labels=c("Infected", "Uninfected"))+
  scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), legend.position= "none", panel.border=element_rect(color="black", fill=NA, size=0.5), axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
Shan

#Simpson diversity
AsyEst_phy_merged_Simpson<- subset(AsyEst_phy_merged, Diversity=="Simpson diversity")

Simp<-ggplot(AsyEst_phy_merged_Simpson, aes(y=Estimator, x=DEC_infection, fill=Infection_diarrhea))+
  geom_boxplot()+
  ylab("Simpson index")+
  stat_compare_means(method="wilcox.test", label.y=32, size=3.5)+
  scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), legend.position= "none", panel.border=element_rect(color="black", fill=NA, size=0.5), axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
Simp

#Observed ASVs
AsyEst_phy_merged_richness<- subset(AsyEst_phy_merged, Diversity=="Species richness")

Obs<-ggplot(AsyEst_phy_merged_richness, aes(y=Estimator, x=DEC_infection, fill=Infection_diarrhea))+
  geom_boxplot()+
  ylab("Observed ASVs")+
  stat_compare_means(method="wilcox.test", label.y=305, size=3.5)+
  scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), legend.position= "none", panel.border=element_rect(color="black", fill=NA, size=0.5), axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
Obs
```

#Boxplots without E. coli
```{r message=FALSE, warning=FALSE}
#shannon diversity
AsyEst_phy_Ec_removed_merged_Shanon<- subset(AsyEst_phy_Ec_removed_merged, Diversity=="Shannon diversity")

Shan_Ec_removed<-ggplot(AsyEst_phy_Ec_removed_merged_Shanon, aes(y=Estimator, x=DEC_infection, fill=Infection_diarrhea))+
  geom_boxplot()+
  ylab("Shannon index")+
  stat_compare_means(method="wilcox.test", label.y=21.5, size=3.5)+
  scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), legend.position= "none", panel.border=element_rect(color="black", fill=NA, size=0.5), axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
Shan_Ec_removed

#Simpson diversity
AsyEst_phy_Ec_removed_merged_Simpson<- subset(AsyEst_phy_Ec_removed_merged, Diversity=="Simpson diversity")

Simp_Ec_removed<-ggplot(AsyEst_phy_Ec_removed_merged_Simpson, aes(y=Estimator, x=DEC_infection, fill=Infection_diarrhea))+
  geom_boxplot()+
  ylab("Simpson index")+
  stat_compare_means(method="wilcox.test", label.y=17, size=3.5)+
  scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), legend.position= "none", panel.border=element_rect(color="black", fill=NA, size=0.5), axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
Simp_Ec_removed

#Observed ASVs
AsyEst_phy_Ec_removed_merged_richness<- subset(AsyEst_phy_Ec_removed_merged, Diversity=="Species richness")

Obs_Ec_removed<-ggplot(AsyEst_phy_Ec_removed_merged_richness, aes(y=Estimator, x=DEC_infection, fill=Infection_diarrhea))+
  geom_boxplot()+
  ylab("Observed ASVs")+
  stat_compare_means(method="wilcox.test", label.y=85, size=3.5)+
  scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), legend.position= "none", panel.border=element_rect(color="black", fill=NA, size=0.5), axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
Obs_Ec_removed
```

#nonpareil (shotgun metagenome data) with E. coli
#import data
```{r message=FALSE, warning=FALSE}
#with E. coli
samples<-read.table("kj_samples_all_human_filtered v2.txt", sep='\t', header =TRUE, as.is=TRUE)%>%
  subset(rotapos=="n")
attach(samples)
```

#create curve objects and summary file for all metagenomes
```{r message=FALSE, warning=FALSE}
#with E. coli
nps<-Nonpareil.set(sample, labels=sample, plot.opts=list(plot.observed=FALSE))

summ<-summary.Nonpareil.Set(nps)
```
#boxplot
```{r message=FALSE, warning=FALSE}
#format data
samples<-samples %>%
  apply(2, str_remove_all, " ") %>%
  as.data.frame()%>%
  'rownames<-'(NULL)%>%
  column_to_rownames(var="sample")

summ_samps<-summ %>%
  as.data.frame()%>%
  merge(samples, by=0)
  
summ_samps$Infection<-factor(summ_samps$Infection, levels=c("Infected", "Uninfected"))

nonpareil<-ggplot(summ_samps, aes(x=Infection, y=diversity, fill=Infection_diarrhea))+
  geom_boxplot()+
  ylab("Nonpareil diversity")+
  scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
  stat_compare_means(method="wilcox.test", label.x=0.8, label.y=18.5, size=3.5)+
  guides(fill=guide_legend(ncol=2))+
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), legend.position= "none", panel.border=element_rect(color="black", fill=NA, size=0.5), axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
nonpareil
```

#test for differences
```{r message=FALSE, warning=FALSE}
#test for normality
shapiro.test(summ_samps$diversity) #data is non-normal (p<0.05)

#DEC infection status
summ_Inf<-subset.data.frame(summ_samps, Infection=="Infected", drop=FALSE)
summ_Uninf<-subset.data.frame(summ_samps, Infection=="Uninfected", drop=FALSE)
wilcox.test(summ_Inf$diversity, summ_Uninf$diversity) 

#diarrhea status
summ_case<-subset.data.frame(summ_samps, Diarrhea=="Case", drop=FALSE)
summ_control<-subset.data.frame(summ_samps, Diarrhea=="Control", drop=FALSE)
wilcox.test(summ_case$diversity, summ_control$diversity) 

#symptomatic v. asymptomati infections
summ_Inf_case<-subset.data.frame(summ_Inf, Diarrhea=="Case", drop=FALSE)
summ_Inf_control<-subset.data.frame(summ_Inf, Diarrhea=="Control", drop=FALSE)
wilcox.test(summ_Inf_case$diversity, summ_Inf_control$diversity) 

#uninfected cases v. controls
summ_Uninf_case<-subset.data.frame(summ_Uninf, Diarrhea=="Case", drop=FALSE)
summ_Uninf_control<-subset.data.frame(summ_Uninf, Diarrhea=="Control", drop=FALSE)
wilcox.test(summ_Uninf_case$diversity, summ_Uninf_control$diversity) 
```

#nonpareil (shotgun metagenome data) without E. coli
#import data
```{r message=FALSE, warning=FALSE}
samples_Ec_removed<-read.table("kj_samples_all_human_filtered_no_Ec v2.txt", sep='\t', header =TRUE, as.is=TRUE)%>%
  subset(rotapos=="n")
attach(samples_Ec_removed)
```

#create curve objects and summary file for all metagenomes
```{r message=FALSE, warning=FALSE}
nps_Ec_removed<-Nonpareil.set(sample, labels=sample, plot.opts=list(plot.observed=FALSE))

summ_Ec_removed<-summary.Nonpareil.Set(nps_Ec_removed)
```
#boxplot
```{r message=FALSE, warning=FALSE}
#format data
samples_Ec_removed<-samples_Ec_removed %>%
  apply(2, str_remove_all, " ") %>%
  as.data.frame()%>%
  'rownames<-'(NULL)%>%
  column_to_rownames(var="sample")

summ_samps_Ec_removed<-summ_Ec_removed %>%
  as.data.frame()%>%
  merge(samples_Ec_removed, by=0)
  
summ_samps_Ec_removed$Infection<-factor(summ_samps_Ec_removed$Infection, levels=c("Infected", "Uninfected"))

nonpareil_Ec_removed<-ggplot(summ_samps_Ec_removed, aes(x=Infection, y=diversity, fill=Infection_diarrhea))+
  geom_boxplot()+
   ylab("Nonpareil diversity")+
  scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
  stat_compare_means(method="wilcox.test", label.x=0.8, label.y=18.5, size=3.5)+
  guides(fill=guide_legend(ncol=2))+
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), legend.position= "none", panel.border=element_rect(color="black", fill=NA, size=0.5), axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
nonpareil_Ec_removed
```

#test for differences
```{r message=FALSE, warning=FALSE}
#test for normality
shapiro.test(summ_samps$diversity) #data is non-normal (p<0.05)

#DEC infection status
summ_Inf<-subset.data.frame(summ_samps, Infection=="Infected", drop=FALSE)
summ_Uninf<-subset.data.frame(summ_samps, Infection=="Uninfected", drop=FALSE)
wilcox.test(summ_Inf$diversity, summ_Uninf$diversity) 

#diarrhea status
summ_case<-subset.data.frame(summ_samps, Diarrhea=="Case", drop=FALSE)
summ_control<-subset.data.frame(summ_samps, Diarrhea=="Control", drop=FALSE)
wilcox.test(summ_case$diversity, summ_control$diversity) 

#symptomatic v. asymptomati infections
summ_Inf_case<-subset.data.frame(summ_Inf, Diarrhea=="Case", drop=FALSE)
summ_Inf_control<-subset.data.frame(summ_Inf, Diarrhea=="Control", drop=FALSE)
wilcox.test(summ_Inf_case$diversity, summ_Inf_control$diversity) 

#uninfected cases v. controls
summ_Uninf_case<-subset.data.frame(summ_Uninf, Diarrhea=="Case", drop=FALSE)
summ_Uninf_control<-subset.data.frame(summ_Uninf, Diarrhea=="Control", drop=FALSE)
wilcox.test(summ_Uninf_case$diversity, summ_Uninf_control$diversity) 
```

#extract legend
```{r message=FALSE, warning=FALSE}
legend_plot<-ggplot(summ_samps_Ec_removed, aes(x=Infection, y=diversity, fill=Infection_diarrhea))+
  geom_boxplot()+
  scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
  stat_compare_means(method="wilcox.test", label.x=0.8, label.y=18, size=3.5)+
  guides(fill=guide_legend(ncol=2))+
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), axis.text.x=element_blank(), axis.title.x=element_blank(), axis.ticks.x=element_blank(), legend.position="bottom")

legend<-get_legend(legend_plot)
l<-as_ggplot(legend)
```


#multiplots
```{r message=FALSE, warning=FALSE}
multi<-plot_grid(Shan, Simp, Obs, nonpareil, ncol=2, labels="AUTO")
multi_legend<-plot_grid(multi, l, ncol=1, rel_heights=c(1, 0.1))
multi_legend

multi_Ec_removed<-plot_grid(Shan_Ec_removed, Simp_Ec_removed, Obs_Ec_removed, nonpareil_Ec_removed, ncol=2, labels="AUTO")
multi_Ec_removed_legend<-plot_grid(multi_Ec_removed, l, ncol=1, rel_heights = c(1,0.1))
multi_Ec_removed_legend
```


