---
title: Processing VFDB results for EcoZUR metagenomes
author: Kelsey Jesser
date: 1/28/2021
---
```{r}
rm(list = ls())
```

#load libraries
```{r message=FALSE}
library(vegan)
library(reshape2)
library(dplyr)
library(kableExtra)
library(ggplot2)
library(gplots)
library(ComplexHeatmap)
library(circlize)
library(stringr)
library(forcats)
library(ggpubr)
library(cowplot)
library(RColorBrewer)
library(tibble)
```

#set theme
```{r}
theme_set(theme_bw())
```

#import data
row names for metadata should be in the same order as column names for VFDB table
```{r}
#metadata
meta<-read.table(file="VFDB_metadata v2.txt", sep='\t', header=TRUE)
rownames(meta)<-NULL

#VFDB relative abundances
VFDB_tab_trans<-read.table(file="VFDB_EcoZUR_MGs_transformed.tsv", sep='\t', header=TRUE, row.names=1, quote="")
```

#filter data
```{r}
#remove samples that are rotavirus-positive or had a contaminated DEC isolate
VFDB_tab_trans<-VFDB_tab_trans%>%
  select(-c("MG_17_bmtagger_filtered_out", "MG_4_bmtagger_filtered_out", "MG_50", "MG_53_CoupledReads_filtered", "MG_54_CoupledReads_filtered", "MG_56_CoupledReads_filtered", "MG_7_bmtagger_filtered_out", "MG_8_bmtagger_filtered_out", "MG_9_bmtagger_filtered_out"))

#filter genes found in <5 metagenomes (~10% of metagenomes)
VFDB_tab_trans<-VFDB_tab_trans[ rowSums(VFDB_tab_trans > 0) >= 10, ]
VFDB_tab_trans_t<-t(VFDB_tab_trans)
```


#subset data by DEC infection and diarrhea (cc) status
```{r}
#metadata
meta_Case<-subset(meta, Diarrhea=="Case" )
meta_Control<-subset(meta, Diarrhea=="Control")
  
meta_Inf<-subset(meta, Infection=="Infected")
meta_Uninf<-subset(meta, Infection=="Uninfected")

meta_Inf_Case<-subset(meta, Infection_diarrhea=="Infected_Case")
meta_Inf_Control<-subset(meta, Infection_diarrhea=="Infected_Control")
meta_Uninf_Case<-subset(meta, Infection_diarrhea=="Uninfected_Case")
meta_Uninf_Control<-subset(meta, Infection_diarrhea=="Uninfected_Control")

#VFDB relative abundance matrix  
Cases<-meta_Case$metagenome_read_file
Controls<-meta_Control$metagenome_read_file
Infected<-meta_Inf$metagenome_read_file
Uninfected<-meta_Uninf$metagenome_read_file
Inf_Cases<-meta_Inf_Case$metagenome_read_file_name
Inf_Controls<-meta_Inf_Control$metagenome_read_file_name
Uninf_Cases <-meta_Uninf_Case$metagenome_read_file_name
Uninf_Controls <-meta_Uninf_Control$metagenome_read_file_name

VFDB_tab_trans_Case<-VFDB_tab_trans%>%
  select(c(Cases))
VFDB_tab_trans_Control<-VFDB_tab_trans%>%
  select(c(Controls))

VFDB_tab_trans_Inf<-VFDB_tab_trans%>%
  select(c(Infected))
VFDB_tab_trans_Uninf<-VFDB_tab_trans%>%
  select(c(Uninfected))

VFDB_tab_trans_Inf_Cases<-VFDB_tab_trans%>%
  select(c(Inf_Cases))
VFDB_tab_trans_Inf_Controls<-VFDB_tab_trans%>%
  select(c(Inf_Controls))
VFDB_tab_trans_Uninf_Cases<-VFDB_tab_trans%>%
  select(c(Uninf_Cases))
VFDB_tab_trans_Uninf_Controls<-VFDB_tab_trans%>%
  select(c(Uninf_Controls))
```

#Kruskal-Wallis testing by DEC infection and diarrhea status
```{r}
#reformat data
VFDB_tab_trans_meta<- meta %>%
  column_to_rownames(var="metagenome_read_file_name")%>%
  merge(VFDB_tab_trans_t, by="row.names")%>%
  dplyr::relocate(Infection, .after=last_col())%>%
  dplyr::relocate(Diarrhea, .after=last_col())%>%
  dplyr::relocate(Infection_diarrhea, .after=last_col())%>%
  select(-c(1:5))

KW_raw_pvalue <- numeric(length = length(1:299))
for (i in (1:299)) {
  KW_raw_pvalue[i] <- kruskal.test(VFDB_tab_trans_meta[, i] ~ VFDB_tab_trans_meta$Infection_diarrhea,
     )$p.value}

KW_df <- data.frame(Variable = names(VFDB_tab_trans_meta[, 1:299]), KW_raw_pvalue = round(KW_raw_pvalue, 8))
KW_df$BH <-p.adjust(KW_df$KW_raw_pvalue,method = "BH")

KW_sig_genes<-subset(KW_df, BH<=0.05)
kbl(KW_sig_genes) %>%
  kable_paper(bootstrap_options = "striped", font_size=10)
```

#format data for heatmaps 
```{r}
#format data
KW_sig_genes<-subset(VFDB_tab_trans, rownames(VFDB_tab_trans) %in% KW_sig_genes$Variable)

KW_sig_genes_mat<-data.matrix(KW_sig_genes)

#calculate relative abundances
KW_sig_genes_inf_case<-KW_sig_genes[,Inf_Cases]
KW_sig_genes_inf_control<-KW_sig_genes[,Inf_Controls]
KW_sig_genes_uninf_case<-KW_sig_genes[,Uninf_Cases]
KW_sig_genes_uninf_control<-KW_sig_genes[,Uninf_Controls]

mean_KW_sig_genes_inf_case<-rowMeans(KW_sig_genes_inf_case)
mean_KW_sig_genes_inf_control<-rowMeans(KW_sig_genes_inf_control)
mean_KW_sig_genes_uninf_case<-rowMeans(KW_sig_genes_uninf_case)
mean_KW_sig_genes_uninf_control<-rowMeans(KW_sig_genes_uninf_control)

mean_KW<-data.frame(inf_case=round(mean_KW_sig_genes_inf_case, 5),
                                inf_control=round(mean_KW_sig_genes_inf_control, 5),
                                uninf_case=round(mean_KW_sig_genes_uninf_case, 5), 
                                uninf_control=round(mean_KW_sig_genes_uninf_control, 5))
mean_KW_mat<-data.matrix(mean_KW)
```

#heatmap
```{r}
#read in streamlined VFDB descriptions
streamlined_KW<-read.csv("mean_KW_mat v2.csv")

streamlined_KW_mat<-streamlined_KW %>%
  subset(select = -c(1:6, 8) ) %>%
  column_to_rownames(var="Function")%>%
  as.matrix()

colnames(streamlined_KW_mat)<-c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls")

#KW
heatmap_annot<-read.csv("mean_heatmap_annot v3.csv")

func_annot<-streamlined_KW%>%
  select(c(7, 8))%>%
  column_to_rownames(var="Function")%>%
  as.matrix()

col = list(Infection_diarrhea=c("Symptomatic DEC infections"="coral2", "Asymptomatic DEC infections"="steelblue3", "Uninfected cases"="darkgoldenrod2", "Uninfected controls"="darkolivegreen4"), Diarrhea = c("Case" = "white", "Control" = "white"), Infection=c("Infected"="white", "Uninfected"="white"))

cn=colnames(streamlined_KW_mat)

levels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls")

ha<-HeatmapAnnotation(Infection_diarrhea=heatmap_annot$Infection_diarrhea, Infection=heatmap_annot$Infection, Diarrhea=heatmap_annot$Diarrhea, col=col, annotation_label=c("Sample group","DEC infection status", "Diarrhea status"))

mycols_mean <- colorRamp2(breaks = c(0,(max(streamlined_KW_mat)/2),max(streamlined_KW_mat)), colors = c("gray98", "mediumpurple", "mediumpurple4"))

h_KW<-Heatmap(streamlined_KW_mat, 
              row_names_gp = gpar(fontsize = 8), 
              show_column_names=FALSE,  
              show_column_dend=FALSE, 
              col=mycols_mean, 
              show_row_dend = FALSE, 
              show_row_names=FALSE,
              column_order=c(1,2,3,4),
              bottom_annotation=ha, 
              heatmap_legend_param=list(title=c("Rel. abund"), legend_height=unit(3, "cm")))

nb.cols<-10
func_color<-order_colors<-colorRampPalette(brewer.pal(10, "Set1"))(nb.cols)
func<-Heatmap(func_annot, 
             width=unit(5,"mm"), 
             show_column_names=FALSE, 
             show_row_names=TRUE, 
             name="Function",
             col=c(func_color),
             row_names_gp=gpar(fontsize=8.5),
             row_names_max_width=max_text_width(rownames(streamlined_KW_mat)))

h_KW+func
```

#Comparison of total number of virulence genes detected by DEC infection and diarrhea status
```{r}
#format data
VFDB_rowsums<-VFDB_tab_trans %>% 
  mutate_if(is.numeric, ~1 * (. !=0)) %>%
  t()%>%
  as.data.frame()%>%
  rownames_to_column(var="metagenome_read_file_name")%>%
  merge(meta, "metagenome_read_file_name")%>%
  dplyr::relocate(sample_name, .after =metagenome_read_file_name)%>%
  column_to_rownames("sample_name")%>%
  mutate(total = rowSums(select_if(., is.numeric), na.rm = TRUE))

VFDB_rowsums_inf_case<-subset(VFDB_rowsums, Infection=="Infected" & Diarrhea=="Case")
VFDB_rowsums_inf_control<-subset(VFDB_rowsums, Infection=="Infected" & Diarrhea=="Control")

VFDB_rowsums_uninf_case<-subset(VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Case")
VFDB_rowsums_uninf_control<-subset(VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Control")

#nonparametric pairwise tests
wilcox.test(VFDB_rowsums_inf_case$total, VFDB_rowsums_inf_control$total)
wilcox.test(VFDB_rowsums_uninf_case$total, VFDB_rowsums_uninf_control$total)
#plots of total virulence genes by DEC infection and diarrhea status

#plot
VFDB_box<-VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
    scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
VFDB_box
```

#Comparison of total number of E. coli-annotated virulence genes detected by DEC infection and diarrhea status
```{r}
#format data
Ec_VFDB_rowsums<-VFDB_tab_trans %>% 
  mutate_if(is.numeric, ~1 * (. !=0)) %>%
  t()%>%
  as.data.frame()%>%
  select(contains("Escherichia"))%>%
  rownames_to_column(var="metagenome_read_file_name")%>%
  merge(meta, "metagenome_read_file_name")%>%
  dplyr::relocate(sample_name, .after =metagenome_read_file_name)%>%
  column_to_rownames("sample_name")%>%
  mutate(total = rowSums(select_if(., is.numeric), na.rm = TRUE))

Ec_VFDB_rowsums_inf_case<-subset(Ec_VFDB_rowsums, Infection=="Infected" & Diarrhea=="Case")
Ec_VFDB_rowsums_inf_control<-subset(Ec_VFDB_rowsums, Infection=="Infected" & Diarrhea=="Control")

Ec_VFDB_rowsums_uninf_case<-subset(Ec_VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Case")
Ec_VFDB_rowsums_uninf_control<-subset(Ec_VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Control")

#nonparametric pairwise tests
wilcox.test(Ec_VFDB_rowsums_inf_case$total, Ec_VFDB_rowsums_inf_control$total)
wilcox.test(Ec_VFDB_rowsums_uninf_case$total, Ec_VFDB_rowsums_uninf_control$total)

#plot
EC_VFDB_box<-Ec_VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
     scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
EC_VFDB_box
```

#Comparison of total number of Klebsiella-annotated virulence genes detected by DEC infection and diarrhea status
```{r}
#format data
Kleb_VFDB_rowsums<-VFDB_tab_trans %>% 
  mutate_if(is.numeric, ~1 * (. !=0)) %>%
  t()%>%
  as.data.frame()%>%
  select(contains("Klebsiella"))%>%
  rownames_to_column(var="metagenome_read_file_name")%>%
  merge(meta, "metagenome_read_file_name")%>%
  dplyr::relocate(sample_name, .after =metagenome_read_file_name)%>%
  column_to_rownames("sample_name")%>%
  mutate(total = rowSums(select_if(., is.numeric), na.rm = TRUE))

Kleb_VFDB_rowsums_inf_case<-subset(Kleb_VFDB_rowsums, Infection=="Infected" & Diarrhea=="Case")
Kleb_VFDB_rowsums_inf_control<-subset(Kleb_VFDB_rowsums, Infection=="Infected" & Diarrhea=="Control")

Kleb_VFDB_rowsums_uninf_case<-subset(Kleb_VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Case")
Kleb_VFDB_rowsums_uninf_control<-subset(Kleb_VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Control")

#nonparametric pairwise tests
wilcox.test(Kleb_VFDB_rowsums_inf_case$total, VFDB_rowsums_inf_control$total)
wilcox.test(Kleb_VFDB_rowsums_uninf_case$total, VFDB_rowsums_uninf_control$total)

#plot
Kleb_VFDB_box<-Kleb_VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
     scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
Kleb_VFDB_box
```

#boxplot multi with Ec result
```{r}
#boxplot
VFDB_box2<-VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
    scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    ggtitle("Total virulence genes (any taxa)")+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), , panel.border=element_rect(color="black", fill=NA, size=0.5), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12), legend.position="none")
VFDB_box2

#boxplot w only E. coli-annotated genes
EC_VFDB_box2<-Ec_VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
    scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    ggtitle(expression(paste(italic("E. coli"),"-annotated virulence genes")))+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
EC_VFDB_box2

box2<-plot_grid(VFDB_box2, EC_VFDB_box2, ncol=1, rel_heights=c(1,1.4), labels=c("A", "B"))
box2
```

