---
title: "Processing VFDB results for EcoZUR metagenomes"
author: "Kelsey Jesser"
date: "7/11/2023"
output: pdf_document
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE) 
```

```{r message=FALSE, warning=FALSE}
rm(list = ls())
```

#Load libraries
```{r message=FALSE}
library(vegan)
library(reshape2)
library(dplyr)
library(kableExtra)
library(ggplot2)
library(gplots)
library(ComplexHeatmap)
library(circlize)
library(stringr)
library(forcats)
library(ggpubr)
library(cowplot)
library(RColorBrewer)
library(tibble)
```

#Set theme
```{r message=FALSE, warning=FALSE}
theme_set(theme_bw())
```

#Import and filter data
row names for metadata should be in the same order as column names for VFDB table
```{r message=FALSE, warning=FALSE}
#metadata
meta<-read.table(file="VFDB_metadata v2.txt", sep='\t', header=TRUE)
rownames(meta)<-NULL

#VFDB relative abundances
VFDB_tab_trans<-read.table(file="VFDB_EcoZUR_MGs_transformed.tsv", sep='\t', header=TRUE, row.names=1, quote="")

#remove samples that are rotavirus-positive or had a contaminated DEC isolate
VFDB_tab_trans<-VFDB_tab_trans%>%
  select(-c("MG_17_bmtagger_filtered_out", "MG_4_bmtagger_filtered_out", "MG_50", "MG_53_CoupledReads_filtered", "MG_54_CoupledReads_filtered", "MG_56_CoupledReads_filtered", "MG_7_bmtagger_filtered_out", "MG_8_bmtagger_filtered_out", "MG_9_bmtagger_filtered_out"))

#filter genes found in <10 metagenomes (~10% of metagenomes)
VFDB_tab_trans<-VFDB_tab_trans[ rowSums(VFDB_tab_trans > 0) >= 10, ]
VFDB_tab_trans_t<-t(VFDB_tab_trans)
```

#Subset data by DEC infection and diarrhea (cc) status
```{r message=FALSE, warning=FALSE}
#metadata
meta_Inf_Case<-subset(meta, Infection_diarrhea=="Infected_Case")
meta_Inf_Control<-subset(meta, Infection_diarrhea=="Infected_Control")
meta_Uninf_Case<-subset(meta, Infection_diarrhea=="Uninfected_Case")
meta_Uninf_Control<-subset(meta, Infection_diarrhea=="Uninfected_Control")

#VFDB relative abundance matrix  
Inf_Cases<-meta_Inf_Case$metagenome_read_file_name
Inf_Controls<-meta_Inf_Control$metagenome_read_file_name
Uninf_Cases <-meta_Uninf_Case$metagenome_read_file_name
Uninf_Controls <-meta_Uninf_Control$metagenome_read_file_name

VFDB_tab_trans_Inf_Cases<-VFDB_tab_trans%>%
  select(c(Inf_Cases))
VFDB_tab_trans_Inf_Controls<-VFDB_tab_trans%>%
  select(c(Inf_Controls))
VFDB_tab_trans_Uninf_Cases<-VFDB_tab_trans%>%
  select(c(Uninf_Cases))
VFDB_tab_trans_Uninf_Controls<-VFDB_tab_trans%>%
  select(c(Uninf_Controls))
```

#Kruskal-Wallis testing by DEC infection and diarrhea status
```{r message=FALSE, warning=FALSE}
#reformat data
VFDB_tab_trans_meta<- meta %>%
  column_to_rownames(var="metagenome_read_file_name")%>%
  merge(VFDB_tab_trans_t, by="row.names")%>%
  dplyr::relocate(Infection, .after=last_col())%>%
  dplyr::relocate(Diarrhea, .after=last_col())%>%
  dplyr::relocate(Infection_diarrhea, .after=last_col())%>%
  select(-c(1:5))

KW_raw_pvalue <- numeric(length = length(1:299))
for (i in (1:299)) {
  KW_raw_pvalue[i] <- kruskal.test(VFDB_tab_trans_meta[, i] ~ VFDB_tab_trans_meta$Infection_diarrhea,
     )$p.value}

KW_df <- data.frame(Variable = names(VFDB_tab_trans_meta[, 1:299]), KW_raw_pvalue = round(KW_raw_pvalue, 8))
KW_df$BH <-p.adjust(KW_df$KW_raw_pvalue,method = "BH")

KW_sig_genes<-subset(KW_df, BH<=0.05)
kbl(KW_sig_genes) %>%
  kable_paper(latex_options = "striped", font_size=10)
```

#Format data for heatmaps 
```{r message=FALSE, warning=FALSE}
#format data
KW_sig_genes<-subset(VFDB_tab_trans, rownames(VFDB_tab_trans) %in% KW_sig_genes$Variable)

KW_sig_genes_mat<-data.matrix(KW_sig_genes)

#calculate relative abundances
KW_sig_genes_inf_case<-KW_sig_genes[,Inf_Cases]
KW_sig_genes_inf_control<-KW_sig_genes[,Inf_Controls]
KW_sig_genes_uninf_case<-KW_sig_genes[,Uninf_Cases]
KW_sig_genes_uninf_control<-KW_sig_genes[,Uninf_Controls]

mean_KW_sig_genes_inf_case<-rowMeans(KW_sig_genes_inf_case)
mean_KW_sig_genes_inf_control<-rowMeans(KW_sig_genes_inf_control)
mean_KW_sig_genes_uninf_case<-rowMeans(KW_sig_genes_uninf_case)
mean_KW_sig_genes_uninf_control<-rowMeans(KW_sig_genes_uninf_control)

mean_KW<-data.frame(inf_case=round(mean_KW_sig_genes_inf_case, 5),
                                inf_control=round(mean_KW_sig_genes_inf_control, 5),
                                uninf_case=round(mean_KW_sig_genes_uninf_case, 5), 
                                uninf_control=round(mean_KW_sig_genes_uninf_control, 5))
mean_KW_mat<-data.matrix(mean_KW)
```

#Heatmap
```{r message=FALSE, warning=FALSE}
#read in streamlined VFDB descriptions
streamlined_KW<-read.csv("mean_KW_mat v2.csv")

streamlined_KW_mat<-streamlined_KW %>%
  subset(select = -c(1:6, 8) ) %>%
  column_to_rownames(var="Function")%>%
  as.matrix()

colnames(streamlined_KW_mat)<-c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls")

#KW
heatmap_annot<-read.csv("mean_heatmap_annot v3.csv")

func_annot<-streamlined_KW%>%
  select(c(7, 8))%>%
  column_to_rownames(var="Function")%>%
  as.matrix()

col = list(Infection_diarrhea=c("Symptomatic DEC infections"="coral2", "Asymptomatic DEC infections"="steelblue3", "Uninfected cases"="darkgoldenrod2", "Uninfected controls"="darkolivegreen4"), Diarrhea = c("Case" = "white", "Control" = "white"), Infection=c("Infected"="white", "Uninfected"="white"))

cn=colnames(streamlined_KW_mat)

levels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls")

ha<-HeatmapAnnotation(Infection_diarrhea=heatmap_annot$Infection_diarrhea, Infection=heatmap_annot$Infection, Diarrhea=heatmap_annot$Diarrhea, col=col, annotation_label=c("Sample group","DEC infection status", "Diarrhea status"))

mycols_mean <- colorRamp2(breaks = c(0,(max(streamlined_KW_mat)/2),max(streamlined_KW_mat)), colors = c("gray98", "mediumpurple", "mediumpurple4"))

h_KW<-Heatmap(streamlined_KW_mat, 
              row_names_gp = gpar(fontsize = 8), 
              show_column_names=FALSE,  
              show_column_dend=FALSE, 
              col=mycols_mean, 
              show_row_dend = FALSE, 
              show_row_names=FALSE,
              column_order=c(1,2,3,4),
              bottom_annotation=ha, 
              heatmap_legend_param=list(title=c("Rel. abund"), legend_height=unit(3, "cm")))

nb.cols<-10
func_color<-order_colors<-colorRampPalette(brewer.pal(10, "Set1"))(nb.cols)
func<-Heatmap(func_annot, 
             width=unit(5,"mm"), 
             show_column_names=FALSE, 
             show_row_names=TRUE, 
             name="Function",
             col=c(func_color),
             row_names_gp=gpar(fontsize=8.5),
             row_names_max_width=max_text_width(rownames(streamlined_KW_mat)))

h_KW+func
```

#Comparison of total number of virulence genes detected by DEC infection and diarrhea status
```{r message=FALSE, warning=FALSE}
#format data
VFDB_rowsums<-VFDB_tab_trans %>% 
  mutate_if(is.numeric, ~1 * (. !=0)) %>%
  t()%>%
  as.data.frame()%>%
  rownames_to_column(var="metagenome_read_file_name")%>%
  merge(meta, "metagenome_read_file_name")%>%
  dplyr::relocate(sample_name, .after =metagenome_read_file_name)%>%
  column_to_rownames("sample_name")%>%
  mutate(total = rowSums(select_if(., is.numeric), na.rm = TRUE))

VFDB_rowsums_inf_case<-subset(VFDB_rowsums, Infection=="Infected" & Diarrhea=="Case")
VFDB_rowsums_inf_control<-subset(VFDB_rowsums, Infection=="Infected" & Diarrhea=="Control")

VFDB_rowsums_uninf_case<-subset(VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Case")
VFDB_rowsums_uninf_control<-subset(VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Control")

#nonparametric pairwise tests
wilcox.test(VFDB_rowsums_inf_case$total, VFDB_rowsums_inf_control$total)
wilcox.test(VFDB_rowsums_uninf_case$total, VFDB_rowsums_uninf_control$total)
#plots of total virulence genes by DEC infection and diarrhea status

#plot
VFDB_box<-VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
    scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
VFDB_box
```

#Comparison of total number of E. coli-annotated virulence genes detected by DEC infection and diarrhea status
```{r message=FALSE, warning=FALSE}
#format data
Ec_VFDB_rowsums<-VFDB_tab_trans %>% 
  mutate_if(is.numeric, ~1 * (. !=0)) %>%
  t()%>%
  as.data.frame()%>%
  select(contains("Escherichia"))%>%
  rownames_to_column(var="metagenome_read_file_name")%>%
  merge(meta, "metagenome_read_file_name")%>%
  dplyr::relocate(sample_name, .after =metagenome_read_file_name)%>%
  column_to_rownames("sample_name")%>%
  mutate(total = rowSums(select_if(., is.numeric), na.rm = TRUE))

Ec_VFDB_rowsums_inf_case<-subset(Ec_VFDB_rowsums, Infection=="Infected" & Diarrhea=="Case")
Ec_VFDB_rowsums_inf_control<-subset(Ec_VFDB_rowsums, Infection=="Infected" & Diarrhea=="Control")

Ec_VFDB_rowsums_uninf_case<-subset(Ec_VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Case")
Ec_VFDB_rowsums_uninf_control<-subset(Ec_VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Control")

#nonparametric pairwise tests
wilcox.test(Ec_VFDB_rowsums_inf_case$total, Ec_VFDB_rowsums_inf_control$total)
wilcox.test(Ec_VFDB_rowsums_uninf_case$total, Ec_VFDB_rowsums_uninf_control$total)

#plot
EC_VFDB_box<-Ec_VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
     scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
EC_VFDB_box
```

#Comparison of total number of Klebsiella-annotated virulence genes detected by DEC infection and diarrhea status
```{r message=FALSE, warning=FALSE}
#format data
Kleb_VFDB_rowsums<-VFDB_tab_trans %>% 
  mutate_if(is.numeric, ~1 * (. !=0)) %>%
  t()%>%
  as.data.frame()%>%
  select(contains("Klebsiella"))%>%
  rownames_to_column(var="metagenome_read_file_name")%>%
  merge(meta, "metagenome_read_file_name")%>%
  dplyr::relocate(sample_name, .after =metagenome_read_file_name)%>%
  column_to_rownames("sample_name")%>%
  mutate(total = rowSums(select_if(., is.numeric), na.rm = TRUE))

Kleb_VFDB_rowsums_inf_case<-subset(Kleb_VFDB_rowsums, Infection=="Infected" & Diarrhea=="Case")
Kleb_VFDB_rowsums_inf_control<-subset(Kleb_VFDB_rowsums, Infection=="Infected" & Diarrhea=="Control")

Kleb_VFDB_rowsums_uninf_case<-subset(Kleb_VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Case")
Kleb_VFDB_rowsums_uninf_control<-subset(Kleb_VFDB_rowsums, Infection=="Uninfected" & Diarrhea=="Control")

#nonparametric pairwise tests
wilcox.test(Kleb_VFDB_rowsums_inf_case$total, VFDB_rowsums_inf_control$total)
wilcox.test(Kleb_VFDB_rowsums_uninf_case$total, VFDB_rowsums_uninf_control$total)

#plot
Kleb_VFDB_box<-Kleb_VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
     scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
Kleb_VFDB_box
```

#boxplot multi with Ec result
```{r message=FALSE, warning=FALSE}
#boxplot
VFDB_box2<-VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
    scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    ggtitle("Total virulence genes (any taxa)")+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), , panel.border=element_rect(color="black", fill=NA, size=0.5), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12), legend.position="none")
VFDB_box2

#boxplot w only E. coli-annotated genes
EC_VFDB_box2<-Ec_VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
    scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    ggtitle(expression(paste(italic("E. coli"),"-annotated virulence genes")))+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
EC_VFDB_box2

box2<-plot_grid(VFDB_box2, EC_VFDB_box2, ncol=1, rel_heights=c(1,1.4), labels=c("A", "B"))
box2
```

#Subset data by age
```{r}
#metadata
ages<-read.csv("ages_metadata_final.csv")%>%
  dplyr::rename(sample_name=Alternate.Sample.ID)%>%
  mutate(sample_name = ifelse(sample_name == "B001", "B1", sample_name))

meta_ages<- meta%>%
  left_join(ages, by=c("sample_name"))%>%
  mutate(Age.months = as.numeric(Age.months),
         Age.years = as.numeric(Age.years))%>%
  dplyr::rename(Diarrhea=Diarrhea.x)%>%
  select(-c(Diarrhea.y))

meta_0to5<-meta_ages%>%
  subset(Age.years<=4)%>%
  subset(Age.years>=0)

#VFDB matrix
VFDB_tab_trans_t_ages<-VFDB_tab_trans_t%>%
  as.data.frame()%>%
  tibble::rownames_to_column("metagenome_read_file_name")%>%
  left_join(meta_ages, by=c("metagenome_read_file_name"))

VFDB_tab_trans_t_0to5<-VFDB_tab_trans_t_ages%>%
  subset(Age.years<=4)%>%
  subset(Age.years>=0)
```

#Subset data by DEC infection and diarrhea status for participants aged <5y
```{r}
meta_0to5_Inf_Case<-subset(meta_0to5, Infection_diarrhea=="Infected_Case")
meta_0to5_Inf_Control<-subset(meta_0to5, Infection_diarrhea=="Infected_Control")
meta_0to5_Uninf_Case<-subset(meta_0to5, Infection_diarrhea=="Uninfected_Case")
meta_0to5_Uninf_Control<-subset(meta_0to5, Infection_diarrhea=="Uninfected_Control")

#VFDB relative abundance matrix  
Inf_Cases_0to5<-meta_0to5_Inf_Case$metagenome_read_file_name
Inf_Controls_0to5<-meta_0to5_Inf_Control$metagenome_read_file_name
Uninf_Cases_0to5<-meta_0to5_Uninf_Case$metagenome_read_file_name
Uninf_Controls_0to5<-meta_0to5_Uninf_Control$metagenome_read_file_name

VFDB_tab_trans_Inf_Cases_0to5<-VFDB_tab_trans_t_0to5%>%
  subset(Infection_diarrhea=="Infected_Case") 
VFDB_tab_trans_Inf_Controls_0to5<-VFDB_tab_trans_t_0to5%>%
  subset(Infection_diarrhea=="Infected_Control") 
VFDB_tab_trans_Uninf_Cases_0to5<-VFDB_tab_trans_t_0to5%>%
  subset(Infection_diarrhea=="Uninfected_Case") 
VFDB_tab_trans_Uninf_Controls_0to5<-VFDB_tab_trans_t_0to5%>%
  subset(Infection_diarrhea=="Uninfected_Control") 
```


#Kruskal-Wallis testing by DEC infection and diarrhea status for participants aged <5y
```{r message=FALSE, warning=FALSE}
#reformat data
VFDB_tab_trans_t_0to5_test<- VFDB_tab_trans_t_0to5%>%
  `rownames<-`( NULL )%>%
  column_to_rownames("metagenome_read_file_name")%>%
  dplyr::relocate(Infection_diarrhea, .after=last_col())%>%
  select(-c(300:316))

KW_raw_pvalue_0to5 <- numeric(length = length(1:299))
for (i in (1:299)) {
  KW_raw_pvalue_0to5[i] <- kruskal.test(VFDB_tab_trans_t_0to5_test[, i] ~ VFDB_tab_trans_t_0to5_test$Infection_diarrhea,
     )$p.value}

KW_df_0to5 <- data.frame(Variable = names(VFDB_tab_trans_t_0to5_test[, 1:299]), KW_raw_pvalue_0to5 = round(KW_raw_pvalue_0to5, 8))
KW_df_0to5$BH <-p.adjust(KW_df_0to5$KW_raw_pvalue_0to5,method = "BH")

KW_sig_genes_0to5<-subset(KW_df_0to5, BH<=0.05)
kbl(KW_sig_genes_0to5) %>%
  kable_paper(latex_options = "striped", font_size=10)
```


#<5y data for heatmap
```{r message=FALSE, warning=FALSE}
#format data
KW_sig_genes_0to5_heat<-subset(VFDB_tab_trans, rownames(VFDB_tab_trans)%in%
                               KW_sig_genes_0to5$Variable)

KW_sig_genes_0to5_heat_mat<-data.matrix(KW_sig_genes_0to5_heat)

#calculate relative abundances
KW_sig_genes_0to5_inf_case<-KW_sig_genes_0to5_heat_mat[,Inf_Cases_0to5]
KW_sig_genes_0to5_inf_control<-KW_sig_genes_0to5_heat_mat[,Inf_Controls_0to5]
KW_sig_genes_0to5_uninf_case<-KW_sig_genes_0to5_heat_mat[,Uninf_Cases_0to5]
KW_sig_genes_0to5_uninf_control<-KW_sig_genes_0to5_heat_mat[,Uninf_Controls_0to5]

mean_KW_sig_genes_0to5_inf_case<-rowMeans(KW_sig_genes_0to5_inf_case)
mean_KW_sig_genes_0to5_inf_control<-rowMeans(KW_sig_genes_0to5_inf_control)
mean_KW_sig_genes_0to5_uninf_case<-rowMeans(KW_sig_genes_0to5_uninf_case)
mean_KW_sig_genes_0to5_uninf_control<-rowMeans(KW_sig_genes_0to5_uninf_control)

mean_KW_0to5<-data.frame(inf_case=round(mean_KW_sig_genes_0to5_inf_case, 5),
                                inf_control=round(mean_KW_sig_genes_0to5_inf_control, 5),
                                uninf_case=round(mean_KW_sig_genes_0to5_uninf_case, 5), 
                                uninf_control=round(mean_KW_sig_genes_0to5_uninf_control, 5))


```

#<5y heatmap
```{r message=FALSE, warning=FALSE}
streamlined_KW_0to5_mat<-streamlined_KW%>%
  as.data.frame()%>%
  select(-c(8:12))%>%
  `rownames<-`( NULL )%>%
  column_to_rownames("VFDB")%>%
  merge(mean_KW_0to5, by=0)%>%
  `rownames<-`( NULL )%>%
  column_to_rownames("Function")%>%
  select(-c(1:6))%>%
  as.matrix()

mycols_mean <- colorRamp2(breaks = c(0,0.10735/2,0.10735), colors = c("gray98", "mediumpurple", "mediumpurple4"))

h_KW_0to5<-Heatmap(streamlined_KW_0to5_mat, 
              row_names_gp = gpar(fontsize = 8), 
              show_column_names=FALSE,  
              show_column_dend=FALSE, 
              col=mycols_mean, 
              show_row_dend = FALSE, 
              show_row_names=FALSE,
              #column_title="Participants aged <5y",
              column_order=c(1,2,3,4),
              bottom_annotation=ha, 
              heatmap_legend_param=list(title=c("Rel. abund"), legend_height=unit(3, "cm")))

func_annot_0to5<-func_annot%>%
  merge(streamlined_KW_0to5_mat, by=0)%>%
  select(-c(3:6))%>%
  column_to_rownames("Row.names")

func_0to5<-Heatmap(func_annot_0to5, 
             width=unit(5,"mm"), 
             show_column_names=FALSE, 
             show_row_names=TRUE, 
             name="Function",
             col=c("#E41A1C", "#4A72A6", "#48A462", "#7E6E85", "#D16948", "#FFB716", "#E1C62F", "#B75F49"),
             row_names_gp=gpar(fontsize=10),
             row_names_max_width=max_text_width(rownames(streamlined_KW_mat)))

h_KW_0to5+func_0to5
```

#Comparison of total number of virulence genes detected by DEC infection and diarrhea status for particpants <5y
```{r message=FALSE, warning=FALSE}
#format data
VFDB_rowsums_0to5<-VFDB_tab_trans_t_0to5 %>% 
  mutate_if(is.numeric, ~1 * (. !=0)) %>%
  dplyr::relocate(sample_name, .after =metagenome_read_file_name)%>%
  remove_rownames()%>%
  column_to_rownames("sample_name")%>%
  mutate(total = rowSums(select_if(., is.numeric), na.rm = TRUE))

VFDB_rowsums_inf_case_0to5<-subset(VFDB_rowsums_0to5, Infection=="Infected" & Diarrhea=="Case")
VFDB_rowsums_inf_control_0to5<-subset(VFDB_rowsums_0to5, Infection=="Infected" & Diarrhea=="Control")

VFDB_rowsums_uninf_case_0to5<-subset(VFDB_rowsums_0to5, Infection=="Uninfected" & Diarrhea=="Case")
VFDB_rowsums_uninf_control_0to5<-subset(VFDB_rowsums_0to5, Infection=="Uninfected" & Diarrhea=="Control")

#nonparametric pairwise tests
wilcox.test(VFDB_rowsums_inf_case_0to5$total, VFDB_rowsums_inf_control_0to5$total)
wilcox.test(VFDB_rowsums_uninf_case_0to5$total, VFDB_rowsums_uninf_control_0to5$total)
#plots of total virulence genes by DEC infection and diarrhea status

#plot
VFDB_box_0to5<-VFDB_rowsums_0to5 %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
    scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
VFDB_box_0to5
```

#Comparison of total number of E. coli-annotated virulence genes detected by DEC infection and diarrhea status for particpants <5y
```{r message=FALSE, warning=FALSE}
#format data
Ec_VFDB_rowsums_0to5 <- VFDB_rowsums_0to5 %>%
  select(contains("Escherichia"))%>%
  rownames_to_column(var="sample_name")%>%
  left_join(meta_0to5, by='sample_name')

Ec_VFDB_rowsums_inf_case_0to5 <-subset(Ec_VFDB_rowsums_0to5 , Infection=="Infected" & Diarrhea=="Case")%>%
  select(-c(1,124:141))%>%
  mutate_all(as.numeric)%>%
  mutate(total = rowSums(.))
Ec_VFDB_rowsums_inf_control_0to5 <-subset(Ec_VFDB_rowsums_0to5 , Infection=="Infected" & Diarrhea=="Control")%>%
  select(-c(1,124:141))%>%
  mutate_all(as.numeric)%>%
  mutate(total = rowSums(.))

Ec_VFDB_rowsums_uninf_case_0to5 <-subset(Ec_VFDB_rowsums_0to5 , Infection=="Uninfected" & Diarrhea=="Case")%>%
  select(-c(1,124:141))%>%
  mutate_all(as.numeric)%>%
  mutate(total = rowSums(.))
Ec_VFDB_rowsums_uninf_control_0to5 <-subset(Ec_VFDB_rowsums_0to5 , Infection=="Uninfected" & Diarrhea=="Control")%>%
  select(-c(1,124:141))%>%
  mutate_all(as.numeric)%>%
  mutate(total = rowSums(.))

#nonparametric pairwise tests
wilcox.test(Ec_VFDB_rowsums_inf_case_0to5$total, Ec_VFDB_rowsums_inf_control_0to5$total)
wilcox.test(Ec_VFDB_rowsums_uninf_case_0to5$total, Ec_VFDB_rowsums_uninf_control_0to5$total)

#plot
EC_VFDB_box_0to5<- Ec_VFDB_rowsums_0to5%>%
  mutate(across(2:123, as.numeric))%>%
  dplyr::mutate(total = rowSums(across(2:123)))%>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
     scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
EC_VFDB_box_0to5
```

#Comparison of total number of Klebsiella-annotated virulence genes detected by DEC infection and diarrhea status for particpants <5y
```{r message=FALSE, warning=FALSE}
#format data
Kleb_VFDB_rowsums_0to5 <- VFDB_rowsums_0to5 %>%
  select(contains("Klebsiella"))%>%
  rownames_to_column(var="sample_name")%>%
  left_join(meta_0to5, by='sample_name')

Kleb_VFDB_rowsums_inf_case_0to5 <-subset(Kleb_VFDB_rowsums_0to5 , Infection=="Infected" & Diarrhea=="Case")%>%
  select(-c(1,83:100))%>%
  mutate_all(as.numeric)%>%
  mutate(total = rowSums(.))
Kleb_VFDB_rowsums_inf_control_0to5 <-subset(Kleb_VFDB_rowsums_0to5 , Infection=="Infected" & Diarrhea=="Control")%>%
  select(-c(1,83:100))%>%
  mutate_all(as.numeric)%>%
  mutate(total = rowSums(.))

Kleb_VFDB_rowsums_uninf_case_0to5 <-subset(Kleb_VFDB_rowsums_0to5 , Infection=="Uninfected" & Diarrhea=="Case")%>%
  select(-c(1,83:100))%>%
  mutate_all(as.numeric)%>%
  mutate(total = rowSums(.))
Kleb_VFDB_rowsums_uninf_control_0to5 <-subset(Kleb_VFDB_rowsums_0to5 , Infection=="Uninfected" & Diarrhea=="Control")%>%
  select(-c(1,83:100))%>%
  mutate_all(as.numeric)%>%
  mutate(total = rowSums(.))

#nonparametric pairwise tests
wilcox.test(Kleb_VFDB_rowsums_inf_case_0to5$total, Kleb_VFDB_rowsums_inf_control_0to5$total)
wilcox.test(Kleb_VFDB_rowsums_uninf_case_0to5$total, Kleb_VFDB_rowsums_uninf_control_0to5$total)

#plot
Kleb_VFDB_box_0to5<-Kleb_VFDB_rowsums_0to5 %>% 
  mutate(across(2:82, as.numeric))%>%
  dplyr::mutate(total = rowSums(across(2:82)))%>%
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
     scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
Kleb_VFDB_box_0to5
```

#boxplot multi with Ec result for participants <5y
```{r message=FALSE, warning=FALSE}
#boxplot
VFDB_box2_0to5<-VFDB_rowsums_0to5   %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
    scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    ggtitle("Total virulence genes (any taxa)")+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), , panel.border=element_rect(color="black", fill=NA, size=0.5), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12), legend.position="none")
VFDB_box2_0to5

#boxplot w only E. coli-annotated genes
EC_VFDB_box2_0to5<-Ec_VFDB_rowsums_0to5%>%
  mutate(across(2:123, as.numeric))%>%
  dplyr::mutate(total = rowSums(across(2:123)))%>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=Infection_diarrhea, y=total, x=Infection))+
    geom_boxplot()+
     scale_fill_manual(values=c("Infected_Case"="coral2", "Infected_Control"="steelblue3", "Uninfected_Case"="darkgoldenrod2", "Uninfected_Control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
EC_VFDB_box2_0to5

box2_0to5<-plot_grid(VFDB_box2_0to5, EC_VFDB_box2_0to5, ncol=1, rel_heights=c(1,1.4), labels=c("A", "B"))
box2_0to5
```