---
title: Analysis of virulence factor database results for EcoZUR metagenomes
author: Kelsey Jesser
date: 1/28/2021
---
```{r message=FALSE, warning=FALSE}
rm(list = ls())
```

#load libraries
```{r message=FALSE, warning=FALSE}
library(vegan)
library(reshape2)
library(dplyr)
library(kableExtra)
library(ggplot2)
library(gplots)
library(ComplexHeatmap)
library(circlize)
library(stringr)
library(forcats)
library(ggpubr)
library(cowplot)
library(RColorBrewer)
library(tibble)
```

#set theme
```{r message=FALSE, warning=FALSE}
theme_set(theme_bw())
```

#import data
row names for metadata should be in the same order as column names for VFDB table
```{r message=FALSE, warning=FALSE}
#metadata
meta<-read.table(file="VFDB_metadata.tsv", sep='\t', header=TRUE)

#VFDB relative abundances
VFDB_tab_trans<-read.table(file="VFDB_EcoZUR_MGs_transformed.tsv", sep='\t', header=TRUE, row.names=1, quote="")
```

#filter rotavirus-positive samples and VFDB genes present in 10 or fewer metagenomes (~10% of metagenomes)
```{r message=FALSE, warning=FALSE}
#remove rotavirus-positive samples
meta<-subset(meta, metagenome_read_file != "MG_17_bmtagger_filtered_out" & metagenome_read_file != "MG_4_bmtagger_filtered_out"& metagenome_read_file != "MG_53_CoupledReads_filtered"& metagenome_read_file != "MG_54_CoupledReads_filtered"& metagenome_read_file != "MG_56_CoupledReads_filtered"& metagenome_read_file != "MG_8_bmtagger_filtered_out"& metagenome_read_file != "MG_9_bmtagger_filtered_out"& metagenome_read_file != "MG_7_bmtagger_filtered_out")
rownames(meta)<-NULL

rota_neg<-c('B101_bmtagger_filtered_out','B147_bmtagger_filtered_out','B259_bmtagger_filtered_out','B293_bmtagger_filtered_out','B312_bmtagger_filtered_out','B329_bmtagger_filtered','B46_bmtagger_filtered_out','C26_bmtagger_filtered_out','C46_bmtagger_filtered_out','C71_bmtagger_filtered_out','E119_bmtagger_filtered_out','E130_bmtagger_filtered_out','E132_bmtagger_filtered_out','E167_bmtagger_filtered2_out','E55_bmtagger_filtered_out','E70_bmtagger_filtered_out','E82_bmtagger_filtered_out','MG_10_bmtagger_filtered_out','MG_11_bmtagger_filtered_out','MG_12_bmtagger_filtered_out','MG_13_bmtagger_filtered_out','MG_14_bmtagger_filtered_out','MG_15_bmtagger_filtered_out','MG_16_bmtagger_filtered_out','MG_18_bmtagger_filtered_out','MG_19_bmtagger_filtered_out','MG_1_bmtagger_filtered_out','MG_20_bmtagger_filtered_out','MG_21_bmtagger_filtered_out','MG_22_bmtagger_filtered_out','MG_23_bmtagger_filtered_out','MG_24_bmtagger_filtered_out','MG_25_bmtagger_filtered_out','MG_26_bmtagger_filtered_out','MG_27_CoupledReads_filtered','MG_28_bmtagger_filtered_out','MG_29_bmtagger_filtered_out','MG_2_CoupledReads_filtered','MG_30_bmtagger_filtered_out','MG_31_human_reads_filtered_out','MG_32_bmtagger_filtered_out','MG_33_bmtagger_filtered_out','MG_34_bmtagger_filtered_out','MG_35_bmtagger_filtered_out','MG_36_bmtagger_filtered_out','MG_37_CoupledReads_filtered','MG_38_CoupledReads_filtered','MG_39_CoupledReads_filtered','MG_3_bmtagger_filtered_out','MG_40_CoupledReads_filtered','MG_41_CoupledReads_filtered','MG_42_CoupledReads_filtered','MG_43_CoupledReads_filtered','MG_44_CoupledReads_filtered','MG_45_CoupledReads_filtered','MG_46_CoupledReads_filtered','MG_47_CoupledReads_filtered','MG_48_CoupledReads_filtered','MG_49_CoupledReads_filtered','MG_50','MG_51_CoupledReads_filtered','MG_52','MG_55_CoupledReads_filtered','MG_57_bmtagger_filtered_out','MG_58_bmtagger_filtered_out','MG_59_CoupledReads_filtered','MG_5_bmtagger_filtered_out','MG_60_CoupledReads_filtered','MG_61_CoupledReads_filtered','MG_62_CoupledReads_filtered','MG_63_CoupledReads_filtered','MG_64_CoupledReads_filtered','MG_65_CoupledReads_filtered','MG_66_CoupledReads_filtered','MG_67_CoupledReads_filtered','MG_68_CoupledReads_filtered','MG_69_CoupledReads_filtered','MG_6_bmtagger_filtered_out','MG_70_CoupledReads_filtered','MG_71_CoupledReads_filtered','MG_72_CoupledReads_filtered','MG_73_CoupledReads_filtered','MG_74_CoupledReads_filtered','MG_75_CoupledReads_filtered','MG_76_CoupledReads_filtered','MG_77_CoupledReads_filtered','MG_78_CoupledReads_filtered','MG_79_CoupledReads_filtered','MG_80_CoupledReads_filtered','MG_81_CoupledReads_filtered','MG_82_CoupledReads_filtered','MG_83_CoupledReads_filtered','MG_84_CoupledReads_filtered','MG_85_CoupledReads_filtered','MG_86_CoupledReads_filtered','MG_87_Coupled_Reads_filtered','MG_88_Coupled_Reads_filtered','Q199_bmtagger_filtered_out','Q312_bmtagger_filtered_out','R119_bmtagger_filtered_out','R55_bmtagger_filtered_out','R84_bmtagger_filtered_out','R85_bmtagger_filtered_out','R8_bmtagger_filtered_out')

VFDB_tab_trans<-VFDB_tab_trans[,rota_neg]

#filter genes found in <10 metagenomes
VFDB_tab_trans<-VFDB_tab_trans[ rowSums(VFDB_tab_trans > 0) >= 10, ]
VFDB_tab_trans_t<-t(VFDB_tab_trans)

```


#subset data by DEC infection and diarrhea (cc) status
```{r message=FALSE, warning=FALSE}
#metadata
meta_pathpos<-subset(meta, Pathotype=="pathotype+")
meta_pathneg<-subset(meta, Pathotype=="pathotype-")
  
meta_case<-subset(meta, Clinical_description=="case")
meta_cont<-subset(meta, Clinical_description=="control")

#Make vectors with sample names (column headers) belonging to each group (make sure order is the same as metadata rowname order)
pathpos<-c('B101_bmtagger_filtered_out','B147_bmtagger_filtered_out','B259_bmtagger_filtered_out','B312_bmtagger_filtered_out','B329_bmtagger_filtered','B46_bmtagger_filtered_out','C26_bmtagger_filtered_out','C46_bmtagger_filtered_out','C71_bmtagger_filtered_out','E119_bmtagger_filtered_out','E130_bmtagger_filtered_out','E132_bmtagger_filtered_out','E167_bmtagger_filtered2_out','E55_bmtagger_filtered_out','E70_bmtagger_filtered_out','MG_10_bmtagger_filtered_out','MG_11_bmtagger_filtered_out','MG_12_bmtagger_filtered_out','MG_13_bmtagger_filtered_out','MG_15_bmtagger_filtered_out','MG_16_bmtagger_filtered_out','MG_18_bmtagger_filtered_out','MG_1_bmtagger_filtered_out','MG_21_bmtagger_filtered_out','MG_22_bmtagger_filtered_out','MG_23_bmtagger_filtered_out','MG_24_bmtagger_filtered_out','MG_25_bmtagger_filtered_out','MG_26_bmtagger_filtered_out','MG_27_CoupledReads_filtered','MG_29_bmtagger_filtered_out','MG_30_bmtagger_filtered_out','MG_31_human_reads_filtered_out','MG_32_bmtagger_filtered_out','MG_33_bmtagger_filtered_out','MG_34_bmtagger_filtered_out','MG_35_bmtagger_filtered_out','MG_3_bmtagger_filtered_out','MG_40_CoupledReads_filtered','MG_41_CoupledReads_filtered','MG_42_CoupledReads_filtered','MG_45_CoupledReads_filtered','MG_46_CoupledReads_filtered','MG_47_CoupledReads_filtered','MG_48_CoupledReads_filtered','MG_51_CoupledReads_filtered','MG_5_bmtagger_filtered_out','MG_6_bmtagger_filtered_out','Q199_bmtagger_filtered_out','Q312_bmtagger_filtered_out','R119_bmtagger_filtered_out','R55_bmtagger_filtered_out','R84_bmtagger_filtered_out','R8_bmtagger_filtered_out')
pathneg<-c('B293_bmtagger_filtered_out','E82_bmtagger_filtered_out','MG_14_bmtagger_filtered_out','MG_19_bmtagger_filtered_out','MG_20_bmtagger_filtered_out','MG_28_bmtagger_filtered_out','MG_2_CoupledReads_filtered','MG_36_bmtagger_filtered_out','MG_37_CoupledReads_filtered','MG_38_CoupledReads_filtered','MG_39_CoupledReads_filtered','MG_43_CoupledReads_filtered','MG_44_CoupledReads_filtered','MG_49_CoupledReads_filtered','MG_50','MG_52','MG_55_CoupledReads_filtered','MG_57_bmtagger_filtered_out','MG_58_bmtagger_filtered_out','MG_59_CoupledReads_filtered','MG_60_CoupledReads_filtered','MG_61_CoupledReads_filtered','MG_62_CoupledReads_filtered','MG_63_CoupledReads_filtered','MG_64_CoupledReads_filtered','MG_65_CoupledReads_filtered','MG_66_CoupledReads_filtered','MG_67_CoupledReads_filtered','MG_68_CoupledReads_filtered','MG_69_CoupledReads_filtered','MG_70_CoupledReads_filtered','MG_71_CoupledReads_filtered','MG_72_CoupledReads_filtered','MG_73_CoupledReads_filtered','MG_74_CoupledReads_filtered','MG_75_CoupledReads_filtered','MG_76_CoupledReads_filtered','MG_77_CoupledReads_filtered','MG_78_CoupledReads_filtered','MG_79_CoupledReads_filtered','MG_80_CoupledReads_filtered','MG_81_CoupledReads_filtered','MG_82_CoupledReads_filtered','MG_83_CoupledReads_filtered','MG_84_CoupledReads_filtered','MG_85_CoupledReads_filtered','MG_86_CoupledReads_filtered','MG_87_Coupled_Reads_filtered','MG_88_Coupled_Reads_filtered','R85_bmtagger_filtered_out')

case<-c('MG_10_bmtagger_filtered_out','MG_11_bmtagger_filtered_out','MG_12_bmtagger_filtered_out','MG_13_bmtagger_filtered_out','MG_14_bmtagger_filtered_out','MG_15_bmtagger_filtered_out','MG_16_bmtagger_filtered_out','MG_18_bmtagger_filtered_out','MG_19_bmtagger_filtered_out','MG_1_bmtagger_filtered_out','MG_20_bmtagger_filtered_out','MG_21_bmtagger_filtered_out','MG_22_bmtagger_filtered_out','MG_23_bmtagger_filtered_out','MG_24_bmtagger_filtered_out','MG_25_bmtagger_filtered_out','MG_26_bmtagger_filtered_out','MG_27_CoupledReads_filtered','MG_28_bmtagger_filtered_out','MG_29_bmtagger_filtered_out','MG_30_bmtagger_filtered_out','MG_31_human_reads_filtered_out','MG_32_bmtagger_filtered_out','MG_33_bmtagger_filtered_out','MG_34_bmtagger_filtered_out','MG_35_bmtagger_filtered_out','MG_36_bmtagger_filtered_out','MG_3_bmtagger_filtered_out','MG_55_CoupledReads_filtered','MG_57_bmtagger_filtered_out','MG_58_bmtagger_filtered_out','MG_59_CoupledReads_filtered','MG_5_bmtagger_filtered_out','MG_6_bmtagger_filtered_out','MG_76_CoupledReads_filtered','MG_77_CoupledReads_filtered','MG_78_CoupledReads_filtered','MG_79_CoupledReads_filtered','MG_80_CoupledReads_filtered','MG_81_CoupledReads_filtered','MG_82_CoupledReads_filtered','MG_83_CoupledReads_filtered','MG_84_CoupledReads_filtered','MG_85_CoupledReads_filtered','MG_86_CoupledReads_filtered','MG_87_Coupled_Reads_filtered','MG_88_Coupled_Reads_filtered')
control<-c('B101_bmtagger_filtered_out','B147_bmtagger_filtered_out','B259_bmtagger_filtered_out','B293_bmtagger_filtered_out','B312_bmtagger_filtered_out','B329_bmtagger_filtered','B46_bmtagger_filtered_out','C26_bmtagger_filtered_out','C46_bmtagger_filtered_out','C71_bmtagger_filtered_out','E119_bmtagger_filtered_out','E130_bmtagger_filtered_out','E132_bmtagger_filtered_out','E167_bmtagger_filtered2_out','E55_bmtagger_filtered_out','E70_bmtagger_filtered_out','E82_bmtagger_filtered_out','MG_2_CoupledReads_filtered','MG_37_CoupledReads_filtered','MG_38_CoupledReads_filtered','MG_39_CoupledReads_filtered','MG_40_CoupledReads_filtered','MG_41_CoupledReads_filtered','MG_42_CoupledReads_filtered','MG_43_CoupledReads_filtered','MG_44_CoupledReads_filtered','MG_45_CoupledReads_filtered','MG_46_CoupledReads_filtered','MG_47_CoupledReads_filtered','MG_48_CoupledReads_filtered','MG_49_CoupledReads_filtered','MG_50','MG_51_CoupledReads_filtered','MG_52','MG_60_CoupledReads_filtered','MG_61_CoupledReads_filtered','MG_62_CoupledReads_filtered','MG_63_CoupledReads_filtered','MG_64_CoupledReads_filtered','MG_65_CoupledReads_filtered','MG_66_CoupledReads_filtered','MG_67_CoupledReads_filtered','MG_68_CoupledReads_filtered','MG_69_CoupledReads_filtered','MG_70_CoupledReads_filtered','MG_71_CoupledReads_filtered','MG_72_CoupledReads_filtered','MG_73_CoupledReads_filtered','MG_74_CoupledReads_filtered','MG_75_CoupledReads_filtered','Q199_bmtagger_filtered_out','Q312_bmtagger_filtered_out','R119_bmtagger_filtered_out','R55_bmtagger_filtered_out','R84_bmtagger_filtered_out','R85_bmtagger_filtered_out','R8_bmtagger_filtered_out')

pathpos_case<-c('MG_10_bmtagger_filtered_out','MG_11_bmtagger_filtered_out','MG_12_bmtagger_filtered_out','MG_13_bmtagger_filtered_out','MG_15_bmtagger_filtered_out','MG_16_bmtagger_filtered_out','MG_18_bmtagger_filtered_out','MG_1_bmtagger_filtered_out','MG_21_bmtagger_filtered_out','MG_22_bmtagger_filtered_out','MG_23_bmtagger_filtered_out','MG_24_bmtagger_filtered_out','MG_25_bmtagger_filtered_out','MG_26_bmtagger_filtered_out','MG_27_CoupledReads_filtered','MG_29_bmtagger_filtered_out','MG_30_bmtagger_filtered_out','MG_31_human_reads_filtered_out','MG_32_bmtagger_filtered_out','MG_33_bmtagger_filtered_out','MG_34_bmtagger_filtered_out','MG_35_bmtagger_filtered_out','MG_3_bmtagger_filtered_out','MG_5_bmtagger_filtered_out','MG_6_bmtagger_filtered_out')
pathpos_control<-c('B101_bmtagger_filtered_out','B147_bmtagger_filtered_out','B259_bmtagger_filtered_out','B312_bmtagger_filtered_out','B329_bmtagger_filtered','B46_bmtagger_filtered_out','C26_bmtagger_filtered_out','C46_bmtagger_filtered_out','C71_bmtagger_filtered_out','E119_bmtagger_filtered_out','E130_bmtagger_filtered_out','E132_bmtagger_filtered_out','E167_bmtagger_filtered2_out','E55_bmtagger_filtered_out','E70_bmtagger_filtered_out','MG_40_CoupledReads_filtered','MG_41_CoupledReads_filtered','MG_42_CoupledReads_filtered','MG_45_CoupledReads_filtered','MG_46_CoupledReads_filtered','MG_47_CoupledReads_filtered','MG_48_CoupledReads_filtered','MG_51_CoupledReads_filtered','Q199_bmtagger_filtered_out','Q312_bmtagger_filtered_out','R119_bmtagger_filtered_out','R55_bmtagger_filtered_out','R85_bmtagger_filtered_out','R8_bmtagger_filtered_out')
pathneg_case<-c('MG_14_bmtagger_filtered_out','MG_19_bmtagger_filtered_out','MG_20_bmtagger_filtered_out','MG_28_bmtagger_filtered_out','MG_36_bmtagger_filtered_out','MG_55_CoupledReads_filtered','MG_57_bmtagger_filtered_out','MG_58_bmtagger_filtered_out','MG_59_CoupledReads_filtered','MG_76_CoupledReads_filtered','MG_77_CoupledReads_filtered','MG_78_CoupledReads_filtered','MG_79_CoupledReads_filtered','MG_80_CoupledReads_filtered','MG_81_CoupledReads_filtered','MG_82_CoupledReads_filtered','MG_83_CoupledReads_filtered','MG_84_CoupledReads_filtered','MG_85_CoupledReads_filtered','MG_86_CoupledReads_filtered','MG_87_Coupled_Reads_filtered','MG_88_Coupled_Reads_filtered')
pathneg_control<-c('B293_bmtagger_filtered_out','E82_bmtagger_filtered_out','MG_2_CoupledReads_filtered','MG_37_CoupledReads_filtered','MG_38_CoupledReads_filtered','MG_39_CoupledReads_filtered','MG_43_CoupledReads_filtered','MG_44_CoupledReads_filtered','MG_49_CoupledReads_filtered','MG_50','MG_52','MG_60_CoupledReads_filtered','MG_61_CoupledReads_filtered','MG_62_CoupledReads_filtered','MG_63_CoupledReads_filtered','MG_64_CoupledReads_filtered','MG_65_CoupledReads_filtered','MG_66_CoupledReads_filtered','MG_67_CoupledReads_filtered','MG_68_CoupledReads_filtered','MG_69_CoupledReads_filtered','MG_70_CoupledReads_filtered','MG_71_CoupledReads_filtered','MG_72_CoupledReads_filtered','MG_73_CoupledReads_filtered','MG_74_CoupledReads_filtered','MG_75_CoupledReads_filtered','R84_bmtagger_filtered_out')

#VFDB relative abundance matrix  
VFDB_tab_trans_pathpos<-VFDB_tab_trans[,pathpos]
VFDB_tab_trans_pathneg<-VFDB_tab_trans[,pathneg]

VFDB_tab_trans_case<-VFDB_tab_trans[,case]
VFDB_tab_trans_cont<-VFDB_tab_trans[,control]
```

#Kruskal-Wallis testing by DEC infection and diarrhea status (sample group)
```{r message=FALSE, warning=FALSE}
#reformat data
VFDB_tab_trans_t<-as.data.frame(VFDB_tab_trans_t)
clin<-meta$Clinical_description 
path<-meta$Pathotype 
names<-meta$metagenome_read_file

VFDB_tab_trans_meta<-cbind(clin, VFDB_tab_trans_t) 
VFDB_tab_trans_meta<-cbind(path, VFDB_tab_trans_meta) 
VFDB_tab_trans_meta<-cbind(names, VFDB_tab_trans_meta) 

VFDB_meta_melt<-VFDB_tab_trans_meta %>%
  melt(measure.vars=c("names")) %>%
  relocate(path, .after=last_col())%>%
  relocate(clin, .after=last_col())

VFDB_meta_melt$cc_path<-str_c(VFDB_meta_melt$cc, "_", VFDB_meta_melt$path)

KW_raw_pvalue <- numeric(length = length(1:303))
for (i in (1:303)) {
  KW_raw_pvalue[i] <- wilcox.test(VFDB_meta_melt[, i] ~ VFDB_meta_melt$cc_path,
     )$p.value}

KW_df <- data.frame(Variable = names(VFDB_meta_melt[, 1:303]), KW_raw_pvalue = round(KW_raw_pvalue, 8))
KW_df$BH <-p.adjust(KW_df$KW_raw_pvalue,method = "BH")

KW_sig_genes<-subset(KW_df, BH<=0.05)
kbl(KW_sig_genes) %>%
  kable_paper(bootstrap_options = "striped", font_size=10)

#for only E. coli genes  
EC_KW_df<-KW_df %>% filter(str_detect(Variable, "Escherichia_coli"))
EC_KW_df$BH_Ec <-p.adjust(EC_KW_df$KW_raw_pvalue,method = "BH")

Ec_KW_sig_genes<-subset(EC_KW_df, BH_Ec<=0.05)
kbl(Ec_KW_sig_genes) %>%
  kable_paper(bootstrap_options = "striped", font_size=10)
```

#format data for plots
```{r message=FALSE, warning=FALSE}
#format data
KW_sig_genes<-subset(VFDB_tab_trans, rownames(VFDB_tab_trans) %in% KW_sig_genes$Variable)
Ec_KW_sig_genes<-subset(VFDB_tab_trans, rownames(VFDB_tab_trans) %in% Ec_KW_sig_genes$Variable)

KW_sig_genes_mat<-data.matrix(KW_sig_genes)
Ec_KW_sig_genes_mat<-data.matrix(Ec_KW_sig_genes)

#calculate relative abundances
KW_sig_genes_pathpos_case<-KW_sig_genes[,pathpos_case]
KW_sig_genes_pathpos_control<-KW_sig_genes[,pathpos_control]
KW_sig_genes_pathneg_case<-KW_sig_genes[,pathneg_case]
KW_sig_genes_pathneg_control<-KW_sig_genes[,pathneg_control]
Ec_KW_sig_genes_pathpos_case<-Ec_KW_sig_genes[,pathpos_case]
Ec_KW_sig_genes_pathpos_control<-Ec_KW_sig_genes[,pathpos_control]
Ec_KW_sig_genes_pathneg_case<-Ec_KW_sig_genes[,pathneg_case]
Ec_KW_sig_genes_pathneg_control<-Ec_KW_sig_genes[,pathneg_control]

mean_KW_sig_genes_pathpos_case<-rowMeans(KW_sig_genes_pathpos_case)
mean_KW_sig_genes_pathpos_control<-rowMeans(KW_sig_genes_pathpos_control)
mean_KW_sig_genes_pathneg_case<-rowMeans(KW_sig_genes_pathneg_case)
mean_KW_sig_genes_pathneg_control<-rowMeans(KW_sig_genes_pathneg_control)
mean_Ec_KW_sig_genes_pathpos_case<-rowMeans(Ec_KW_sig_genes_pathpos_case)
mean_Ec_KW_sig_genes_pathpos_control<-rowMeans(Ec_KW_sig_genes_pathpos_control)
mean_Ec_KW_sig_genes_pathneg_case<-rowMeans(Ec_KW_sig_genes_pathneg_case)
mean_Ec_KW_sig_genes_pathneg_control<-rowMeans(Ec_KW_sig_genes_pathneg_control)

mean_KW<-data.frame(pathpos_case=round(mean_KW_sig_genes_pathpos_case, 5),
                                pathpos_control=round(mean_KW_sig_genes_pathpos_control, 5),
                                pathneg_case=round(mean_KW_sig_genes_pathneg_case, 5), 
                                pathneg_control=round(mean_KW_sig_genes_pathneg_control, 5))
mean_KW_mat<-data.matrix(mean_KW)

mean_Ec_KW<-data.frame(pathpos_case=round(mean_Ec_KW_sig_genes_pathpos_case, 5),
                                pathpos_control=round(mean_Ec_KW_sig_genes_pathpos_control, 5),
                                pathneg_case=round(mean_Ec_KW_sig_genes_pathneg_case, 5), 
                                pathneg_control=round(mean_Ec_KW_sig_genes_pathneg_control, 5))
mean_Ec_KW_mat<-data.matrix(mean_Ec_KW)
```

#Comparison of total number of virulence genes detected by DEC infection and diarrhea status
```{r message=FALSE, warning=FALSE}
#convert VFDB dataframe to presence/absence
VFDB_PA<-VFDB_tab_trans %>% 
  mutate_if(is.numeric, ~1 * (. !=0)) 
colnames(VFDB_PA)<-meta$sample_name
VFDB_PA<-t(VFDB_PA)


#new dataframes with rowsums
VFDB_rowsums<-data.frame(rowSums=rowSums(VFDB_PA), cc=meta$Clin, path=meta$Pathotype, cc_path=meta$Pathotype_Clinical)

VFDB_rowsums_case<-subset(VFDB_rowsums, cc=="case")
VFDB_rowsums_control<-subset(VFDB_rowsums, cc=="control")

VFDB_rowsums_pathpos<-subset(VFDB_rowsums, path=="pathotype+")
VFDB_rowsums_pathneg<-subset(VFDB_rowsums, path=="pathotype-")

VFDB_rowsums_pathpos_case<-subset(VFDB_rowsums, path=="pathotype+" & cc=="case")
VFDB_rowsums_pathpos_control<-subset(VFDB_rowsums, path=="pathotype+" & cc=="control")

VFDB_rowsums_pathneg_case<-subset(VFDB_rowsums, path=="pathotype-" & cc=="case")
VFDB_rowsums_pathneg_control<-subset(VFDB_rowsums, path=="pathotype-" & cc=="control")

#nonparametric pairwise tests
wilcox.test(VFDB_rowsums_case$rowSums, VFDB_rowsums_control$rowSums)
wilcox.test(VFDB_rowsums_pathpos$rowSums, VFDB_rowsums_pathneg$rowSums)
wilcox.test(VFDB_rowsums_pathpos_case$rowSums, VFDB_rowsums_pathpos_control$rowSums)
wilcox.test(VFDB_rowsums_pathneg_case$rowSums, VFDB_rowsums_pathneg_control$rowSums)
```
#Compare total number of E. coli-annotated virulence genes detected by DEC infection and diarrhea status
```{r message=FALSE, warning=FALSE}
#subset E. coli genes
EC_VFDB_PA<-VFDB_PA %>%
  as.data.frame() %>%
  select(+contains("Escherichia_coli")) %>%
  mutate_if(is.numeric, ~1 * (. !=0))

#new dataframes with rowsums
EC_VFDB_rowsums<-data.frame(rowSums=rowSums(EC_VFDB_PA), cc=meta$Clin, path=meta$Pathotype, cc_path=meta$Pathotype_Clinical)

EC_VFDB_rowsums_case<-subset(EC_VFDB_rowsums, cc=="case")
EC_VFDB_rowsums_control<-subset(EC_VFDB_rowsums, cc=="control")

EC_VFDB_rowsums_pathpos<-subset(EC_VFDB_rowsums, path=="pathotype+")
EC_VFDB_rowsums_pathneg<-subset(EC_VFDB_rowsums, path=="pathotype-")

EC_VFDB_rowsums_pathpos_case<-subset(EC_VFDB_rowsums, path=="pathotype+" & cc=="case")
EC_VFDB_rowsums_pathpos_control<-subset(EC_VFDB_rowsums, path=="pathotype+" & cc=="control")

EC_VFDB_rowsums_pathneg_case<-subset(EC_VFDB_rowsums, path=="pathotype-" & cc=="case")
EC_VFDB_rowsums_pathneg_control<-subset(EC_VFDB_rowsums, path=="pathotype-" & cc=="control")

#nonparametric pairwise tests
wilcox.test(EC_VFDB_rowsums_case$rowSums, EC_VFDB_rowsums_control$rowSums)
wilcox.test(EC_VFDB_rowsums_pathpos$rowSums, EC_VFDB_rowsums_pathneg$rowSums)
wilcox.test(EC_VFDB_rowsums_pathpos_case$rowSums, EC_VFDB_rowsums_pathpos_control$rowSums)
wilcox.test(EC_VFDB_rowsums_pathneg_case$rowSums, EC_VFDB_rowsums_pathneg_control$rowSums)
```
#Compare total number of Klebsiella-annotated virulence genes detected by DEC infection and diarrhea status
```{r message=FALSE, warning=FALSE}
#subset Klebsiella genes
Kleb_VFDB_PA<-VFDB_PA %>%
  as.data.frame() %>%
  select(+contains("Klebsiella")) %>%
  mutate_if(is.numeric, ~1 * (. !=0))

#new dataframes with rowsums
Kleb_VFDB_rowsums<-data.frame(rowSums=rowSums(Kleb_VFDB_PA), cc=meta$Clin, path=meta$Pathotype, cc_path=meta$Pathotype_Clinical)

Kleb_VFDB_rowsums_case<-subset(Kleb_VFDB_rowsums, cc=="case")
Kleb_VFDB_rowsums_control<-subset(Kleb_VFDB_rowsums, cc=="control")

Kleb_VFDB_rowsums_pathpos<-subset(Kleb_VFDB_rowsums, path=="pathotype+")
Kleb_VFDB_rowsums_pathneg<-subset(Kleb_VFDB_rowsums, path=="pathotype-")

Kleb_VFDB_rowsums_pathpos_case<-subset(Kleb_VFDB_rowsums, path=="pathotype+" & cc=="case")
Kleb_VFDB_rowsums_pathpos_control<-subset(Kleb_VFDB_rowsums, path=="pathotype+" & cc=="control")

Kleb_VFDB_rowsums_pathneg_case<-subset(Kleb_VFDB_rowsums, path=="pathotype-" & cc=="case")
Kleb_VFDB_rowsums_pathneg_control<-subset(Kleb_VFDB_rowsums, path=="pathotype-" & cc=="control")

#nonparametric pairwise tests
wilcox.test(Kleb_VFDB_rowsums_case$rowSums, Kleb_VFDB_rowsums_control$rowSums)
wilcox.test(Kleb_VFDB_rowsums_pathpos$rowSums, Kleb_VFDB_rowsums_pathneg$rowSums)
wilcox.test(Kleb_VFDB_rowsums_pathpos_case$rowSums, Kleb_VFDB_rowsums_pathpos_control$rowSums)
wilcox.test(Kleb_VFDB_rowsums_pathneg_case$rowSums, Kleb_VFDB_rowsums_pathneg_control$rowSums)

```

#plot of total virulence genes by DEC infection and diarrhea status
```{r message=FALSE, warning=FALSE}
VFDB_rowsums$path <- factor(VFDB_rowsums$path, levels=c("pathotype+", "pathotype-"))

VFDB_path_box<-VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=cc_path, y=rowSums, x=path))+
    geom_boxplot()+
    scale_fill_manual(values=c("pathotype+_case"="coral2", "pathotype+_control"="steelblue3", "pathotype-_case"="darkgoldenrod2", "pathotype-_control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    #ggtitle("All samples")+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
VFDB_path_box
```

#plot of E.coli virulence genes by cc and pathotype status
```{r message=FALSE, warning=FALSE}
EC_VFDB_rowsums$path <- factor(EC_VFDB_rowsums$path, levels=c("pathotype+", "pathotype-"))

EC_VFDB_path_box<-EC_VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=cc_path, y=rowSums, x=path))+
    geom_boxplot()+
    scale_fill_manual(values=c("pathotype+_case"="coral2", "pathotype+_control"="steelblue3", "pathotype-_case"="darkgoldenrod2", "pathotype-_control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    #ggtitle("All samples")+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
EC_VFDB_path_box

```
#plot of Klebsiella virulence genes by cc and pathotype status
```{r message=FALSE, warning=FALSE}
Kleb_VFDB_rowsums$path <- factor(Kleb_VFDB_rowsums$path, levels=c("pathotype+", "pathotype-"))

Kleb_VFDB_path_box<-Kleb_VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=cc_path, y=rowSums, x=path))+
    geom_boxplot()+
    scale_fill_manual(values=c("pathotype+_case"="coral2", "pathotype+_control"="steelblue3", "pathotype-_case"="darkgoldenrod2", "pathotype-_control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    #ggtitle("All samples")+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
Kleb_VFDB_path_box
```

#boxplot multi with Ec result
```{r message=FALSE, warning=FALSE}
#boxplot
VFDB_rowsums$path <- factor(VFDB_rowsums$path, levels=c("pathotype+", "pathotype-"))

VFDB_path_box2<-VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=cc_path, y=rowSums, x=path))+
    geom_boxplot()+
    scale_fill_manual(values=c("pathotype+_case"="coral2", "pathotype+_control"="steelblue3", "pathotype-_case"="darkgoldenrod2", "pathotype-_control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    ggtitle("Total virulence genes (any taxa)")+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), , panel.border=element_rect(color="black", fill=NA, size=0.5), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12), legend.position="none")
VFDB_path_box2

#boxplot w only E. coli-annotated genes
EC_VFDB_rowsums$path <- factor(EC_VFDB_rowsums$path, levels=c("pathotype+", "pathotype-"))

EC_VFDB_path_box2<-EC_VFDB_rowsums %>% 
  tibble::rownames_to_column("Sample_ID") %>%
  ggplot(aes(fill=cc_path, y=rowSums, x=path))+
    geom_boxplot()+
    scale_fill_manual(values=c("pathotype+_case"="coral2", "pathotype+_control"="steelblue3", "pathotype-_case"="darkgoldenrod2", "pathotype-_control"="darkolivegreen4"), labels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls"))+
    ylab("Number of genes")+
    xlab(NULL)+
    stat_compare_means(method="wilcox.test", label.x=0.8, size=3.5)+
    guides(fill=guide_legend(nrow=2), byrow=TRUE)+
    ggtitle(expression(paste(italic("E. coli"),"-annotated virulence genes")))+
    theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), legend.title=element_blank(), panel.border=element_rect(color="black", fill=NA, size=0.5), legend.position="bottom", axis.text.x=element_blank(), axis.ticks.x=element_blank(), legend.text=element_text(size=12), axis.text.y=element_text(size=10), axis.title.y=element_text(size=12))
EC_VFDB_path_box2

box2<-plot_grid(VFDB_path_box2, EC_VFDB_path_box2, ncol=1, rel_heights=c(1,1.3), labels=c("A", "B"))
box2
```

#heatmap
```{r message=FALSE, warning=FALSE}
#read in streamlined VFDB descriptions
streamlined_KW<-read.csv("mean_KW_mat.csv")

streamlined_KW_mat<-streamlined_KW %>%
  subset(select = -c(1:7) ) %>%
  column_to_rownames("description")%>%
  as.matrix()

colnames(streamlined_KW_mat)<-c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls")

#KW
heatmap_annot<-read.csv("mean_heatmap_annot v2.csv")

org_annot<-read.csv("mean_KW_mat_descriptors_org.csv") %>%
  column_to_rownames(var="description")%>%
  as.matrix()

func_annot<-read.csv("mean_KW_mat_descriptors_func.csv") %>%
  column_to_rownames(var="description")%>%
  as.matrix()

operon_annot<-read.csv("mean_KW_mat_descriptors_operon.csv") %>%
  column_to_rownames(var="description")%>%
  as.matrix()

col = list(pathotype_cc=c("Symptomatic DEC infections"="coral2", "Asymptomatic DEC infections"="steelblue3", "Uninfected cases"="darkgoldenrod2", "Uninfected controls"="darkolivegreen4" ),
           path=c("Pathotype-negative"="white", "Pathotype-positive"="white"),
           cc=c("Case"="White", "Control"="White"))

cn=colnames(streamlined_KW_mat)

levels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls")

ha<-HeatmapAnnotation(pathotype_cc=heatmap_annot$pathotype_cc,
                      cc=heatmap_annot$cc,
                      path=heatmap_annot$pathotype,
                      col=col, 
                      annotation_label=c("Sample group", "DEC infection status", "Diarrhea status"), 
                      #text=anno_text(cn,just="center", location=0.5),
                      #gp=gpar(col="white", border="black"),
                      annotation_height=(max_text_width(cn)))

mycols_mean <- colorRamp2(breaks = c(0,(max(streamlined_KW_mat)/2),max(streamlined_KW_mat)), colors = c("gray98", "mediumpurple", "mediumpurple4"))

h_KW<-Heatmap(streamlined_KW_mat, 
              row_names_gp = gpar(fontsize = 8), 
              show_column_names=FALSE,  
              show_column_dend=FALSE, 
              col=mycols_mean, 
              show_row_dend = FALSE, 
              show_row_names=FALSE,
              column_order=c(1,2,3,4),
              bottom_annotation=ha, 
              heatmap_legend_param=list(title=c("Rel. abund"), legend_height=unit(3, "cm")))
#h_KW<-draw(h_KW, heatmap_legend_side="left", annotation_legend_side="left")

org<-Heatmap(org_annot, 
             width=unit(5,"mm"), 
             show_column_names=FALSE, 
             show_row_names=FALSE, 
             name="Organism",
             col=c("lightsteelblue2", "Gray60", "lightskyblue3", "lightskyblue4", "gray80"))

nb.cols<-9
func_color<-order_colors<-colorRampPalette(rev(brewer.pal(9, "Paired")))(nb.cols)
func<-Heatmap(func_annot, 
             width=unit(5,"mm"), 
             show_column_names=FALSE, 
             show_row_names=TRUE, 
             name="Function",
             col=c(func_color),
             row_names_gp=gpar(fontsize=9),
             row_names_max_width=max_text_width(rownames(streamlined_KW_mat)))


nb.cols<-23
operon_color<-order_colors<-colorRampPalette(brewer.pal(23, "Spectral"))(nb.cols)
operon<-Heatmap(operon_annot, 
           width=unit(5,"mm"), 
           show_column_names=FALSE, 
           show_row_names=TRUE, 
           name="Operon", 
           row_names_max_width=max_text_width(rownames(mean_KW_mat)),
           col=c(operon_color))

h_KW+func

```

