---
title: "16S heatmaps"
author: "Kelsey Jesser"
date: "12/1/2021"
output: html_document
---
#clear R env
```{r include=FALSE}
rm(list = ls())
```

#load libraries
```{r include=FALSE}
library(lefser)
require(dplyr)
require(tibble)
library(qiime2R)
library(circlize)
library(ggplot2)
library(microbiomeMarker)
library(RColorBrewer)
library(ggthemr)
library(plyr)
library(phyloseq)
library(tidyverse)
library(ComplexHeatmap)
library(microViz)
library(cowplot)
library(corncob)
```

#set theme
```{r}
ggthemr('fresh')
```

#convert qiime2 output to phyloseq object
```{r include=FALSE} 
phy<-qza_to_phyloseq("table_EcoZUR.qza", 
                     "rooted_tree_masked_aligned_rep_set_EcoZUR.qza", 
                     "taxonomy_EcoZUR.qza", 
                     "EcoZUR_meta_updated_pathotypes.tsv")
phy
```

#filter dataset
```{r include=FALSE}
#remove taxa where phylum is ambiguous (these are likely artifacts)
phy <- subset_taxa(phy, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))

#remove chloroplast and mitochondrial sequences
phy<-subset_taxa(phy, (Class!="Cloroplast") | is.na(Class))
phy<-subset_taxa(phy, (Family!="mitochondria") | is.na(Class))

#remove taxa that occur <2x and in <10% of samples
phy = filter_taxa(phy, function(x) sum(x > 2) > (0.1*length(x)), TRUE)

#remove rotavirus-positive samples
phy<- subset_samples(phy, rota!="1")

meta<-read.csv(file = "16S_metadata_DEC_diarrhea.csv")
phy<-ps_join(x = phy, y = meta, type = "inner", match_sample_names="SampleID")

#glom at the family level
phy_fam <-phy%>%
  tax_glom("Family")

```

#run corncob to test for differences in relative abundance, adjusting for age category and sex on abundance and age category, sex, and cc + pathotype on variability
```{r}
d <- differentialTest(formula = ~Infection_diarrhea+age_cat+sex,
                                 phi.formula = ~Infection_diarrhea+age_cat+sex,
                                 formula_null = ~age_cat+sex,
                                 phi.formula_null = ~Infection_diarrhea+age_cat+sex,
                                 data = phy_fam,
                                 test = "Wald", boot = FALSE,
                                 fdr_cutoff = .05)

plot(d, level="Family")

sig_marker_corncob<-as.data.frame(otu_to_taxonomy(OTU = d$significant_taxa, data = phy_fam))
```

#run lefse
```{r}
Lefse <- run_lefse(
    phy_fam,
    wilcoxon_cutoff = 0.05,
    group = "Infection_diarrhea",
    kw_cutoff = 0.05,
    multigrp_strat = TRUE,
    lda_cutoff = 3,
    taxa_rank="Family"
)
Lefse

sig_marker_lefse<-data.frame(marker_table(Lefse))

plot_abundance(Lefse, group="Infection_diarrhea")
plot_ef_bar(Lefse)
```


#plot boxplots of significant taxa
```{r}
phy_fam<-phy_fam %>%
  transform_sample_counts(function(x) x / sum(x) )

phy_fam_sig<-subset_taxa(phy_fam, Family=="[Paraprevotellaceae]" | Family=="Bacteroidaceae" | Family == "Verrucomicrobiaceae"| Family == "Bifidobacteriaceae" | Family == "Clostridiaceae"  | Family == "Erysipelotrichaceae" | Family == "Enterobacteriaceae" | Family == "Porphyromonadaceae" | Family == "Methanobacteriaceae" | Family == "Pasteurellaceae" | Family == "Fusobacteriaceae" | Family == "Lachnospiraceae" | Family == "Rikenellaceae" | Family== "Eubacteriaceae" | Family == "Elusimicrobiaceae") 
taxa_names(phy_fam_sig) <- tax_table(phy_fam_sig)[ ,"Family"]

sample_data(phy_fam_sig)$Infection_diarrhea<-factor(sample_data(phy_fam_sig)$Infection_diarrhea, levels=c("Infected_Case", "Infected_Control", "Uninfected_Case", "Uninfected_Control"))

a<-psmelt(phy_fam_sig) %>%
  ggplot(data = ., aes(x = Infection_diarrhea, y = Abundance, fill=NA)) +
    geom_jitter(aes(color = Infection_diarrhea), height = 0, width = .4) +
    geom_boxplot(outlier.shape  = NA) +
    labs(x = "", y = "Abundance\n") +
    facet_wrap(~ OTU, scales = "free")+
    theme(panel.grid.major = element_blank(), axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(),legend.title=element_blank(), legend.position=c(0.9,0.1))+
    scale_color_manual(values =c("coral2", "steelblue3", "darkgoldenrod2", "darkolivegreen4"), labels= c("Symptptomatic\nDEC infections", "Asymptomatic\nDEC infections", "Uninfected cases", "Uninfected controls"))

a
```

#sqrt transformed boxplots
```{r}
phy_fam_sig_transformed <-transform_sample_counts(phy_fam_sig, function(x) sqrt(x))

sample_data(phy_fam_sig_transformed)$Infection_diarrhea<-factor(sample_data(phy_fam_sig)$Infection_diarrhea, levels=c("Infected_Case", "Infected_Control", "Uninfected_Case", "Uninfected_Control"))

b<-psmelt(phy_fam_sig_transformed) %>%
  ggplot(data = ., aes(x = Infection_diarrhea, y = Abundance, fill=NA)) +
    geom_jitter(aes(color = Infection_diarrhea), height = 0, width = .4, alpha=0.75) +
    geom_boxplot(outlier.shape  = NA) +
    labs(x = "", y = "Relative abundance\n") +
    facet_wrap(~ OTU, scales = "free")+
    theme(panel.grid.major = element_blank(), axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(),legend.title=element_blank(), legend.position=c(0.9,0.1))+
  scale_color_manual(values =c("coral2", "steelblue3", "darkgoldenrod2", "darkolivegreen4"), labels= c("Symptptomatic\nDEC infections", "Asymptomatic\nDEC infections", "Uninfected cases", "Uninfected controls"))
b
```

#Heatmap
```{r}
taxa_names(phy_fam_sig) <- tax_table(phy_fam_sig)[ ,"Family"]
sample_data<-phy_fam_sig %>%
  sample_data() %>%
  as.data.frame()%>%
  rownames_to_column()

samp<-data.frame(sample_data$rowname, sample_data$Infection_diarrhea)
samp<-na.omit(samp)
names(samp)[1] <- "rowname"

rel_abund<-otu_table(phy_fam_sig, taxa_are_rows = TRUE)
rel_abund_t<-rel_abund %>%
  t()%>%
  as.data.frame()%>%
  tibble::rownames_to_column()

rel_abund_join<-merge(rel_abund_t, samp, by="rowname")

rel_abund_Infected_Case<-rel_abund_join%>%
    subset(sample_data.Infection_diarrhea == "Infected_Case")%>%
    subset(select=-c(1,17))
rel_abund_Infected_Control<-rel_abund_join%>%
  subset(sample_data.Infection_diarrhea == "Infected_Control")%>%
    subset(select=-c(1,17))
rel_abund_Uninfected_Case<-rel_abund_join%>%
    subset(sample_data.Infection_diarrhea == "Uninfected_Case")%>%
    subset(select=-c(1,17))
rel_abund_Uninfected_Control<-rel_abund_join%>%
  subset(sample_data.Infection_diarrhea == "Uninfected_Control")%>%
    subset(select=-c(1,17))

mean_Infected_Case<-colMeans(rel_abund_Infected_Case)
mean_Infected_Control<-colMeans(rel_abund_Infected_Control)
mean_Uninfected_Case<-colMeans(rel_abund_Uninfected_Case)
mean_Uninfected_Control<-colMeans(rel_abund_Uninfected_Control)

mean_rel_abunds<-data.frame(
  Infected_Case=round(mean_Infected_Case, 5),
  Infected_Control=round(mean_Infected_Control, 5),
  Uninfected_Case=round(mean_Uninfected_Case, 5), 
  Uninfected_Control=round(mean_Uninfected_Control, 5))
mean_rel_abunds_mat<-data.matrix(mean_rel_abunds)

heatmap_annot<-read.csv("mean_heatmap_annot v3.csv")
stat_annot<-read.csv("corncob_lefse_annot.csv") %>%
  column_to_rownames(var="X")%>%
  as.data.frame.matrix() 

col = list(Infection_diarrhea=c("Symptomatic DEC infections"="coral2", "Asymptomatic DEC infections"="steelblue3", "Uninfected cases"="darkgoldenrod2", "Uninfected controls"="darkolivegreen4"), Diarrhea = c("Case" = "white", "Control" = "white"), Infection=c("Infected"="white", "Uninfected"="white"))

levels=c("Symptomatic DEC infections", "Asymptomatic DEC infections", "Uninfected cases", "Uninfected controls")

ha<-HeatmapAnnotation(Infection_diarrhea=heatmap_annot$Infection_diarrhea, Infection=heatmap_annot$Infection, Diarrhea=heatmap_annot$Diarrhea, col=col, annotation_label=c("Sample group","DEC infection status", "Diarrhea status"))

mycols_mean <- colorRamp2(breaks = c(0,(max(mean_rel_abunds_mat)/2),max(mean_rel_abunds_mat)), colors = c("gray98", "mediumpurple", "mediumpurple4"))
h_KW<-Heatmap(mean_rel_abunds_mat, 
              row_names_gp = gpar(fontsize = 12), 
              show_column_names=FALSE,  
              show_column_dend=FALSE,
              column_order = c("Infected_Case", "Infected_Control", "Uninfected_Case", "Uninfected_Control"),
              col=mycols_mean, 
              show_row_dend = FALSE, 
              row_names_max_width=max_text_width(rownames(mean_rel_abunds_mat)), 
              bottom_annotation=ha,   
              #right_annotation=ha2,
              #rect_gp=gpar(col="white", lwd=2),
              heatmap_legend_param=list(title=c("Rel. abund"), legend_height=unit(3, "cm")))

stat<-Heatmap(stat_annot, col=c("lightpink1", "darkslategray3"), width=unit(5, "mm"), name="Method", show_column_names = FALSE, show_row_names = TRUE, heatmap_legend_param=list(labels=c("LEfSe", "Corncob+LEfSe")))

h_KW+stat
```

#sqrt-transformed heatmap
```{r}
taxa_names(phy_fam_sig_transformed) <- tax_table(phy_fam_sig_transformed)[ ,"Family"]
sample_data<-phy_fam_sig_transformed %>%
  sample_data() %>%
  as.data.frame()%>%
  rownames_to_column()

samp<-data.frame(sample_data$rowname, sample_data$Infection_diarrhea)
samp<-na.omit(samp)
names(samp)[1] <- "rowname"

rel_abund<-otu_table(phy_fam_sig_transformed, taxa_are_rows = TRUE)
rel_abund_t<-rel_abund %>%
  t()%>%
  as.data.frame()%>%
  tibble::rownames_to_column()

rel_abund_join<-merge(rel_abund_t, samp, by="rowname")

rel_abund_Infected_Case<-rel_abund_join%>%
    subset(sample_data.Infection_diarrhea == "Infected_Case")%>%
    subset(select=-c(1,17))
rel_abund_Infected_Control<-rel_abund_join%>%
  subset(sample_data.Infection_diarrhea == "Infected_Control")%>%
    subset(select=-c(1,17))
rel_abund_Uninfected_Case<-rel_abund_join%>%
    subset(sample_data.Infection_diarrhea == "Uninfected_Case")%>%
    subset(select=-c(1,17))
rel_abund_Uninfected_Control<-rel_abund_join%>%
  subset(sample_data.Infection_diarrhea == "Uninfected_Control")%>%
    subset(select=-c(1,17))

mean_Infected_Case<-colMeans(rel_abund_Infected_Case)
mean_Infected_Control<-colMeans(rel_abund_Infected_Control)
mean_Uninfected_Case<-colMeans(rel_abund_Uninfected_Case)
mean_Uninfected_Control<-colMeans(rel_abund_Uninfected_Control)

mean_rel_abunds<-data.frame(
  Infected_Case=round(mean_Infected_Case, 5),
  Infected_Control=round(mean_Infected_Control, 5),
  Uninfected_Case=round(mean_Uninfected_Case, 5), 
  Uninfected_Control=round(mean_Uninfected_Control, 5))
mean_rel_abunds_mat<-data.matrix(mean_rel_abunds)

heatmap_annot<-read.csv("mean_heatmap_annot v3.csv")
stat_annot<-read.csv("corncob_lefse_annot.csv") %>%
  column_to_rownames(var="X")%>%
  as.data.frame.matrix()

col = list(Infection_diarrhea=c("Symptomatic DEC infections"="coral2", "Asymptomatic DEC infections"="steelblue3", "Uninfected cases"="darkgoldenrod2", "Uninfected controls"="darkolivegreen4"), Diarrhea = c("Case" = "white", "Control" = "white"), Infection=c("Infected"="white", "Uninfected"="white"))

ha<-HeatmapAnnotation(Infection_diarrhea=heatmap_annot$Infection_diarrhea, Infection=heatmap_annot$Infection, Diarrhea=heatmap_annot$Diarrhea, col=col, annotation_label=c("Sample group","DEC infection status", "Diarrhea status"))

mycols_mean <- colorRamp2(breaks = c(0,(max(mean_rel_abunds_mat)/2),max(mean_rel_abunds_mat)), colors = c("gray98", "mediumpurple", "mediumpurple4"))
h_KW<-Heatmap(mean_rel_abunds_mat, 
              row_names_gp = gpar(fontsize = 12), 
              show_column_names=FALSE,  
              show_column_dend=FALSE,
              column_order = c("Infected_Case", "Infected_Control", "Uninfected_Case", "Uninfected_Control"),
              col=mycols_mean, 
              show_row_dend = FALSE, 
              row_names_max_width=max_text_width(rownames(mean_rel_abunds_mat)), 
              bottom_annotation=ha, 
              #right_annotation=ha2,
              #rect_gp=gpar(col="white", lwd=2),
              heatmap_legend_param=list(title=c("Sqrt rel. abund"), legend_height=unit(3, "cm")))

stat<-Heatmap(stat_annot, col=c("lightpink1", "darkslategray3"), width=unit(5, "mm"), name="Method", show_column_names = FALSE, show_row_names = TRUE, heatmap_legend_param=list(labels=c("LEfSe", "Corncob+LEfSe"))) 
j_KW<-h_KW+stat
j_KW
```

#mean relative abundance values
```{r}
#pathpos control-enriched taxa
mean(rel_abund_Infected_Control$Bifidobacteriaceae)
mean(rel_abund_Infected_Control$Rikenellaceae)
mean(rel_abund_Infected_Control$Ruminococcaceae)
mean(rel_abund_Infected_Control$Verrucomicrobiaceae)
mean(rel_abund_Infected_Control$Bacteroidaceae)

#abundances of other pathpos control-enriched taxa in other groups
rel_abund_other<-rel_abund_join%>%
    subset(sample_data.Infection_diarrhea == "Infected_Case" | sample_data.Infection_diarrhea == "Uninfected_Case" |sample_data.Infection_diarrhea == "Uninfected_Control")%>%
    subset(select=-c(1,17))

mean(rel_abund_other$Bifidobacteriaceae)
mean(rel_abund_other$Rikenellaceae)
mean(rel_abund_other$Ruminococcaceae)
mean(rel_abund_other$Verrucomicrobiaceae)
mean(rel_abund_other$Bacteroidaceae)
```
